<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Patent Citations | The WIPO Patent Analytics Handbook</title>
  <meta name="description" content="Chapter 6 Patent Citations | The WIPO Patent Analytics Handbook." />
  <meta name="generator" content="bookdown 0.28.1 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Patent Citations | The WIPO Patent Analytics Handbook" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Chapter 6 Patent Citations | The WIPO Patent Analytics Handbook." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Patent Citations | The WIPO Patent Analytics Handbook" />
  
  <meta name="twitter:description" content="Chapter 6 Patent Citations | The WIPO Patent Analytics Handbook." />
  

<meta name="author" content="Paul Oldham" />


<meta name="date" content="2022-01-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="classification.html"/>
<link rel="next" href="textmining.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/d3-4.13.0/d3.min.js"></script>
<script src="libs/sankey-1/sankey.js"></script>
<script src="libs/sankeyNetwork-binding-0.4/sankeyNetwork.js"></script>
<link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-1.3.1/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<script src="libs/proj4-2.6.2/proj4.min.js"></script>
<script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script>
<link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-binding-2.1.1/leaflet.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/sund2b-binding-2.1.6/sund2b.js"></script>
<link href="libs/d2b-1.0.9/d2b_custom.css" rel="stylesheet" />
<script src="libs/d2b-1.0.9/d2b.min.js"></script>
<link href="libs/collapsibleTree-0.1.6/collapsibleTree.css" rel="stylesheet" />
<script src="libs/collapsibleTree-binding-0.1.7/collapsibleTree.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The WIPO Patent Analytics Handbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="" data-path="about-the-author.html"><a href="about-the-author.html"><i class="fa fa-check"></i>About the Author</a></li>
<li class="chapter" data-level="" data-path="acknowlegements.html"><a href="acknowlegements.html"><i class="fa fa-check"></i>Acknowlegements</a></li>
<li class="chapter" data-level="" data-path="how-to-use-the-handbook.html"><a href="how-to-use-the-handbook.html"><i class="fa fa-check"></i>How to use the Handbook</a></li>
<li class="chapter" data-level="" data-path="note-to-readers.html"><a href="note-to-readers.html"><i class="fa fa-check"></i>Note to Readers</a></li>
<li class="chapter" data-level="" data-path="datasets.html"><a href="datasets.html"><i class="fa fa-check"></i>Datasets</a></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="literature.html"><a href="literature.html"><i class="fa fa-check"></i><b>2</b> Scientific Literature</a>
<ul>
<li class="chapter" data-level="2.1" data-path="literature.html"><a href="literature.html#accessing-the-scientific-literature"><i class="fa fa-check"></i><b>2.1</b> Accessing the Scientific Literature</a></li>
<li class="chapter" data-level="2.2" data-path="literature.html"><a href="literature.html#searching-literature-databases"><i class="fa fa-check"></i><b>2.2</b> Searching Literature Databases</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="literature.html"><a href="literature.html#stemming"><i class="fa fa-check"></i><b>2.2.1</b> Stemming</a></li>
<li class="chapter" data-level="2.2.2" data-path="literature.html"><a href="literature.html#using-search-operators"><i class="fa fa-check"></i><b>2.2.2</b> Using Search Operators</a></li>
<li class="chapter" data-level="2.2.3" data-path="literature.html"><a href="literature.html#proximity-operators"><i class="fa fa-check"></i><b>2.2.3</b> Proximity Operators</a></li>
<li class="chapter" data-level="2.2.4" data-path="literature.html"><a href="literature.html#regular-expressions"><i class="fa fa-check"></i><b>2.2.4</b> Regular Expressions</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="literature.html"><a href="literature.html#precision-vs.-recall"><i class="fa fa-check"></i><b>2.3</b> Precision vs. Recall</a></li>
<li class="chapter" data-level="2.4" data-path="literature.html"><a href="literature.html#processing-scientific-literature"><i class="fa fa-check"></i><b>2.4</b> Processing Scientific Literature</a></li>
<li class="chapter" data-level="2.5" data-path="literature.html"><a href="literature.html#visualizing-the-scientific-literature"><i class="fa fa-check"></i><b>2.5</b> Visualizing the Scientific Literature</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="literature.html"><a href="literature.html#dashboards"><i class="fa fa-check"></i><b>2.5.1</b> Dashboards</a></li>
<li class="chapter" data-level="2.5.2" data-path="literature.html"><a href="literature.html#network-visualisation"><i class="fa fa-check"></i><b>2.5.2</b> Network Visualisation</a></li>
<li class="chapter" data-level="2.5.3" data-path="literature.html"><a href="literature.html#other-forms-of-visualisation"><i class="fa fa-check"></i><b>2.5.3</b> Other forms of visualisation</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="literature.html"><a href="literature.html#linking-the-scientific-literature-with-patent-analysis"><i class="fa fa-check"></i><b>2.6</b> Linking the Scientific Literature with Patent Analysis</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="literature.html"><a href="literature.html#mapping-authors-to-inventors"><i class="fa fa-check"></i><b>2.6.1</b> Mapping Authors to Inventors</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="literature.html"><a href="literature.html#linking-citations-with-patent-literature"><i class="fa fa-check"></i><b>2.7</b> Linking Citations with Patent Literature</a></li>
<li class="chapter" data-level="2.8" data-path="literature.html"><a href="literature.html#conclusion"><i class="fa fa-check"></i><b>2.8</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="geocoding.html"><a href="geocoding.html"><i class="fa fa-check"></i><b>3</b> Geocoding</a>
<ul>
<li class="chapter" data-level="3.1" data-path="geocoding.html"><a href="geocoding.html#getting-started"><i class="fa fa-check"></i><b>3.1</b> Getting Started</a></li>
<li class="chapter" data-level="3.2" data-path="geocoding.html"><a href="geocoding.html#getting-set-up-with-the-google-maps-api"><i class="fa fa-check"></i><b>3.2</b> Getting set up with the Google Maps API</a></li>
<li class="chapter" data-level="3.3" data-path="geocoding.html"><a href="geocoding.html#using-the-api"><i class="fa fa-check"></i><b>3.3</b> Using the API</a></li>
<li class="chapter" data-level="3.4" data-path="geocoding.html"><a href="geocoding.html#the-source-data"><i class="fa fa-check"></i><b>3.4</b> The Source Data</a></li>
<li class="chapter" data-level="3.5" data-path="geocoding.html"><a href="geocoding.html#lookup-the-records"><i class="fa fa-check"></i><b>3.5</b> Lookup the Records</a></li>
<li class="chapter" data-level="3.6" data-path="geocoding.html"><a href="geocoding.html#using-placement"><i class="fa fa-check"></i><b>3.6</b> Using placement</a></li>
<li class="chapter" data-level="3.7" data-path="geocoding.html"><a href="geocoding.html#using-ggmap"><i class="fa fa-check"></i><b>3.7</b> Using ggmap</a></li>
<li class="chapter" data-level="3.8" data-path="geocoding.html"><a href="geocoding.html#using-googleway"><i class="fa fa-check"></i><b>3.8</b> Using Googleway</a></li>
<li class="chapter" data-level="3.9" data-path="geocoding.html"><a href="geocoding.html#reviewing-initial-results"><i class="fa fa-check"></i><b>3.9</b> Reviewing Initial Results</a></li>
<li class="chapter" data-level="3.10" data-path="geocoding.html"><a href="geocoding.html#tackling-abbreviations"><i class="fa fa-check"></i><b>3.10</b> Tackling Abbreviations</a></li>
<li class="chapter" data-level="3.11" data-path="geocoding.html"><a href="geocoding.html#lookup-edited-names"><i class="fa fa-check"></i><b>3.11</b> Lookup edited names</a></li>
<li class="chapter" data-level="3.12" data-path="geocoding.html"><a href="geocoding.html#bringing-the-data-together"><i class="fa fa-check"></i><b>3.12</b> Bringing the data together</a></li>
<li class="chapter" data-level="3.13" data-path="geocoding.html"><a href="geocoding.html#assessing-the-quality-of-geocoding"><i class="fa fa-check"></i><b>3.13</b> Assessing the Quality of Geocoding</a></li>
<li class="chapter" data-level="3.14" data-path="geocoding.html"><a href="geocoding.html#preprocess-the-data-and-rerun-the-query"><i class="fa fa-check"></i><b>3.14</b> Preprocess the Data and Rerun the Query</a></li>
<li class="chapter" data-level="3.15" data-path="geocoding.html"><a href="geocoding.html#duplicated-affiliation-names"><i class="fa fa-check"></i><b>3.15</b> Duplicated Affiliation Names</a></li>
<li class="chapter" data-level="3.16" data-path="geocoding.html"><a href="geocoding.html#mapping-the-data"><i class="fa fa-check"></i><b>3.16</b> Mapping the Data</a></li>
<li class="chapter" data-level="3.17" data-path="geocoding.html"><a href="geocoding.html#round-up"><i class="fa fa-check"></i><b>3.17</b> Round Up</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="patents.html"><a href="patents.html"><i class="fa fa-check"></i><b>4</b> Counting Patent Data</a>
<ul>
<li class="chapter" data-level="4.1" data-path="patents.html"><a href="patents.html#the-structure-of-patent-numbers"><i class="fa fa-check"></i><b>4.1</b> The structure of patent numbers</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="patents.html"><a href="patents.html#the-country-code"><i class="fa fa-check"></i><b>4.1.1</b> The country code</a></li>
<li class="chapter" data-level="4.1.2" data-path="patents.html"><a href="patents.html#the-numeric-identifier"><i class="fa fa-check"></i><b>4.1.2</b> The numeric identifier</a></li>
<li class="chapter" data-level="4.1.3" data-path="patents.html"><a href="patents.html#kind-codes"><i class="fa fa-check"></i><b>4.1.3</b> Kind Codes</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="patents.html"><a href="patents.html#preparing-to-count-patent-data"><i class="fa fa-check"></i><b>4.2</b> Preparing to Count Patent Data</a></li>
<li class="chapter" data-level="4.3" data-path="patents.html"><a href="patents.html#understanding-priority-numbers"><i class="fa fa-check"></i><b>4.3</b> Understanding Priority Numbers</a></li>
<li class="chapter" data-level="4.4" data-path="patents.html"><a href="patents.html#counting-priority-applications"><i class="fa fa-check"></i><b>4.4</b> Counting Priority Applications</a></li>
<li class="chapter" data-level="4.5" data-path="patents.html"><a href="patents.html#counting-applications"><i class="fa fa-check"></i><b>4.5</b> Counting Applications</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="patents.html"><a href="patents.html#mapping-publications-family-members"><i class="fa fa-check"></i><b>4.5.1</b> Mapping Publications (Family Members)</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="patents.html"><a href="patents.html#trends-by-country-using-publication-data"><i class="fa fa-check"></i><b>4.6</b> Trends by Country using Publication Data</a></li>
<li class="chapter" data-level="4.7" data-path="patents.html"><a href="patents.html#families"><i class="fa fa-check"></i><b>4.7</b> Patent Families</a></li>
<li class="chapter" data-level="4.8" data-path="patents.html"><a href="patents.html#modelling-data"><i class="fa fa-check"></i><b>4.8</b> Modelling Data</a>
<ul>
<li class="chapter" data-level="4.8.1" data-path="patents.html"><a href="patents.html#the-data"><i class="fa fa-check"></i><b>4.8.1</b> The Data</a></li>
<li class="chapter" data-level="4.8.2" data-path="patents.html"><a href="patents.html#loess-smoothing"><i class="fa fa-check"></i><b>4.8.2</b> Loess Smoothing</a></li>
</ul></li>
<li class="chapter" data-level="4.9" data-path="patents.html"><a href="patents.html#forecasting"><i class="fa fa-check"></i><b>4.9</b> Forecasting</a>
<ul>
<li class="chapter" data-level="4.9.1" data-path="patents.html"><a href="patents.html#day-of-days"><i class="fa fa-check"></i><b>4.9.1</b> Day of Days</a></li>
<li class="chapter" data-level="4.9.2" data-path="patents.html"><a href="patents.html#the-data-1"><i class="fa fa-check"></i><b>4.9.2</b> The Data</a></li>
</ul></li>
<li class="chapter" data-level="4.10" data-path="patents.html"><a href="patents.html#conclusion-1"><i class="fa fa-check"></i><b>4.10</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="classification.html"><a href="classification.html"><i class="fa fa-check"></i><b>5</b> Patent Classfication</a>
<ul>
<li class="chapter" data-level="5.1" data-path="classification.html"><a href="classification.html#exploring-the-international-patent-classification"><i class="fa fa-check"></i><b>5.1</b> Exploring the International Patent Classification</a></li>
<li class="chapter" data-level="5.2" data-path="classification.html"><a href="classification.html#the-us-ipc-table"><i class="fa fa-check"></i><b>5.2</b> The US IPC Table</a></li>
<li class="chapter" data-level="5.3" data-path="classification.html"><a href="classification.html#assessing-relationships-between-technology-areas"><i class="fa fa-check"></i><b>5.3</b> Assessing Relationships Between Technology Areas</a></li>
<li class="chapter" data-level="5.4" data-path="classification.html"><a href="classification.html#visualising-relationships-with-chord-diagrams"><i class="fa fa-check"></i><b>5.4</b> Visualising Relationships with Chord Diagrams</a></li>
<li class="chapter" data-level="5.5" data-path="classification.html"><a href="classification.html#the-short-ipc"><i class="fa fa-check"></i><b>5.5</b> The Short IPC</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="classification.html"><a href="classification.html#case-study-finding-a-needle-in-a-haystack"><i class="fa fa-check"></i><b>5.5.1</b> Case Study: Finding A Needle in A Haystack</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="classification.html"><a href="classification.html#classification-and-patent-overlay-mapping"><i class="fa fa-check"></i><b>5.6</b> Classification and Patent Overlay Mapping</a></li>
<li class="chapter" data-level="5.7" data-path="classification.html"><a href="classification.html#the-structure-of-patent-activity"><i class="fa fa-check"></i><b>5.7</b> The Structure of Patent Activity</a></li>
<li class="chapter" data-level="5.8" data-path="classification.html"><a href="classification.html#conclusion-2"><i class="fa fa-check"></i><b>5.8</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="citations.html"><a href="citations.html"><i class="fa fa-check"></i><b>6</b> Patent Citations</a>
<ul>
<li class="chapter" data-level="6.1" data-path="citations.html"><a href="citations.html#non-patent-literature"><i class="fa fa-check"></i><b>6.1</b> Non Patent Literature</a></li>
<li class="chapter" data-level="6.2" data-path="citations.html"><a href="citations.html#literature-and-patent-citation-data-with-the-lens"><i class="fa fa-check"></i><b>6.2</b> Literature and Patent Citation Data with the Lens</a></li>
<li class="chapter" data-level="6.3" data-path="citations.html"><a href="citations.html#retrieving-citations-at-scale-with-patcite"><i class="fa fa-check"></i><b>6.3</b> Retrieving Citations at Scale with PATCITE</a></li>
<li class="chapter" data-level="6.4" data-path="citations.html"><a href="citations.html#the-us-patentsview-non-patent-literature-table"><i class="fa fa-check"></i><b>6.4</b> The US PatentsView Non-Patent Literature Table</a></li>
<li class="chapter" data-level="6.5" data-path="citations.html"><a href="citations.html#patent-citations"><i class="fa fa-check"></i><b>6.5</b> Patent Citations</a></li>
<li class="chapter" data-level="6.6" data-path="citations.html"><a href="citations.html#navigating-patent-networks"><i class="fa fa-check"></i><b>6.6</b> Navigating Patent Networks</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="citations.html"><a href="citations.html#back-citations"><i class="fa fa-check"></i><b>6.6.1</b> Back citations</a></li>
<li class="chapter" data-level="6.6.2" data-path="citations.html"><a href="citations.html#forward-citations"><i class="fa fa-check"></i><b>6.6.2</b> Forward Citations</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="citations.html"><a href="citations.html#counting-citations-by-patent-families"><i class="fa fa-check"></i><b>6.7</b> Counting Citations by Patent Families</a></li>
<li class="chapter" data-level="6.8" data-path="citations.html"><a href="citations.html#patent-citations-by-generation"><i class="fa fa-check"></i><b>6.8</b> Patent Citations by Generation</a></li>
<li class="chapter" data-level="6.9" data-path="citations.html"><a href="citations.html#citations-and-knowledge-spillovers"><i class="fa fa-check"></i><b>6.9</b> Citations and Knowledge Spillovers</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="textmining.html"><a href="textmining.html"><i class="fa fa-check"></i><b>7</b> Text Mining</a>
<ul>
<li class="chapter" data-level="7.1" data-path="textmining.html"><a href="textmining.html#the-uspto-patent-granted-data"><i class="fa fa-check"></i><b>7.1</b> The USPTO Patent Granted Data</a></li>
<li class="chapter" data-level="7.2" data-path="textmining.html"><a href="textmining.html#words"><i class="fa fa-check"></i><b>7.2</b> Words</a></li>
<li class="chapter" data-level="7.3" data-path="textmining.html"><a href="textmining.html#removing-stop-words"><i class="fa fa-check"></i><b>7.3</b> Removing Stop Words</a></li>
<li class="chapter" data-level="7.4" data-path="textmining.html"><a href="textmining.html#lemmatizing-words"><i class="fa fa-check"></i><b>7.4</b> Lemmatizing words</a></li>
<li class="chapter" data-level="7.5" data-path="textmining.html"><a href="textmining.html#terms-by-technology"><i class="fa fa-check"></i><b>7.5</b> Terms by Technology</a></li>
<li class="chapter" data-level="7.6" data-path="textmining.html"><a href="textmining.html#combining-text-mining-with-patent-classification"><i class="fa fa-check"></i><b>7.6</b> Combining Text Mining with Patent Classification</a></li>
<li class="chapter" data-level="7.7" data-path="textmining.html"><a href="textmining.html#from-words-to-phrases-ngrams"><i class="fa fa-check"></i><b>7.7</b> From words to phrases (ngrams)</a></li>
<li class="chapter" data-level="7.8" data-path="textmining.html"><a href="textmining.html#terms-over-time"><i class="fa fa-check"></i><b>7.8</b> Terms over time</a></li>
<li class="chapter" data-level="7.9" data-path="textmining.html"><a href="textmining.html#correlation-measures"><i class="fa fa-check"></i><b>7.9</b> Correlation Measures</a></li>
<li class="chapter" data-level="7.10" data-path="textmining.html"><a href="textmining.html#conclusion-3"><i class="fa fa-check"></i><b>7.10</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="machinelearning.html"><a href="machinelearning.html"><i class="fa fa-check"></i><b>8</b> Machine Learning</a>
<ul>
<li class="chapter" data-level="8.1" data-path="machinelearning.html"><a href="machinelearning.html#from-dictionaries-to-word-vectors"><i class="fa fa-check"></i><b>8.1</b> From Dictionaries to Word Vectors</a></li>
<li class="chapter" data-level="8.2" data-path="machinelearning.html"><a href="machinelearning.html#word-vectors-with-fasttext"><i class="fa fa-check"></i><b>8.2</b> Word Vectors with fastText</a></li>
<li class="chapter" data-level="8.3" data-path="machinelearning.html"><a href="machinelearning.html#training-word-vectors-for-drones"><i class="fa fa-check"></i><b>8.3</b> Training Word Vectors for Drones</a></li>
<li class="chapter" data-level="8.4" data-path="machinelearning.html"><a href="machinelearning.html#using-word-vectors"><i class="fa fa-check"></i><b>8.4</b> Using Word Vectors</a></li>
<li class="chapter" data-level="8.5" data-path="machinelearning.html"><a href="machinelearning.html#exploring-analogies"><i class="fa fa-check"></i><b>8.5</b> Exploring Analogies</a></li>
<li class="chapter" data-level="8.6" data-path="machinelearning.html"><a href="machinelearning.html#patent-specific-word-embeddings"><i class="fa fa-check"></i><b>8.6</b> Patent Specific Word Embeddings</a></li>
<li class="chapter" data-level="8.7" data-path="machinelearning.html"><a href="machinelearning.html#machine-learning-in-classification"><i class="fa fa-check"></i><b>8.7</b> Machine learning in Classification</a></li>
<li class="chapter" data-level="8.8" data-path="machinelearning.html"><a href="machinelearning.html#machine-learning-for-text-classification"><i class="fa fa-check"></i><b>8.8</b> Machine Learning for Text Classification</a>
<ul>
<li class="chapter" data-level="8.8.1" data-path="machinelearning.html"><a href="machinelearning.html#step-1-binary-text-classification"><i class="fa fa-check"></i><b>8.8.1</b> Step 1: Binary Text Classification</a></li>
<li class="chapter" data-level="8.8.2" data-path="machinelearning.html"><a href="machinelearning.html#step-2-multilabel-text-classification"><i class="fa fa-check"></i><b>8.8.2</b> Step 2: Multilabel Text Classification</a></li>
<li class="chapter" data-level="8.8.3" data-path="machinelearning.html"><a href="machinelearning.html#step-3-named-entity-recognition"><i class="fa fa-check"></i><b>8.8.3</b> Step 3: Named Entity Recognition</a></li>
</ul></li>
<li class="chapter" data-level="8.9" data-path="machinelearning.html"><a href="machinelearning.html#from-vector-based-models-to-transformers"><i class="fa fa-check"></i><b>8.9</b> From Vector Based Models to Transformers</a></li>
<li class="chapter" data-level="8.10" data-path="machinelearning.html"><a href="machinelearning.html#conclusion-4"><i class="fa fa-check"></i><b>8.10</b> Conclusion</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="literature.html"><a href="literature.html#conclusion"><i class="fa fa-check"></i><b>9</b> Conclusion</a></li>
<li class="divider"></li>
<li><a href="https://github.com/wipo-analytics/handbook" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The WIPO Patent Analytics Handbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="citations" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Patent Citations<a href="citations.html#citations" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<!--- last spell checked 2022-08-18--->
<p>This chapter focuses on the use of patent citations in patent analytics. Patent citations take two main forms:</p>
<ol style="list-style-type: lower-alpha">
<li>Citations of the scientific literature and other material such as news articles and websites, known as the Non-Patent Literature or NPL</li>
<li>Citations of patent documents.</li>
</ol>
<p>The patent citation system is similar to the familiar academic citation system. However, patent citations differ from academic citations because they <em>limit the scope</em> of what an applicant can claim to be new or novel or as involving an inventive step. As the American economist Suzanne Scotchmer reminds us when approaching the patent system applicants are ‘standing on the shoulders of giants’ <span class="citation">(<a href="#ref-Scotchmer_1991" role="doc-biblioref">Scotchmer 1991</a>)</span>. Put simply new applicants are confronted by the combined weight of the scientific literature, other material and existing patent applications that make up the <em>prior art</em>. The prior art limits the scope of what may be claimed by new applicants and is recorded in patent citations.</p>
<p>Citations within the patent system have become an important focus for research in fields such as econometrics, scientometrics and innovation studies. Citations of the non-patent literature are important focus of research because they help to reveal the closeness of the relationship between scientific research and innovative activity reflected in the patent system. Citations of patent documents are a focus of research because the number of citations that a patent document or patent family attracts is an indicator of social and economic value <span class="citation">(<a href="#ref-Jaffe_2017" role="doc-biblioref">A. B. Jaffe and Rassenfosse 2017</a>)</span>. In addition, analysis of patent citations can lead to the identification of similar patent documents in a technology field, technology spillovers and technology trajectories.</p>
<p>A rich literature has emerged around patent citation analysis and we will highlight some of the key sources in the course of this chapter. Recent work by <span class="citation">A. B. Jaffe and Rassenfosse (<a href="#ref-Jaffe_2017" role="doc-biblioref">2017</a>)</span> provides an accessible and detailed overview of social scientific research involving patent citations. For those seeking to ground themselves in key literature the work by Adam Jaffe and Manuel Trajtenberg <span class="citation">(<a href="#ref-Jaffe_1998" role="doc-biblioref">A. Jaffe and Trajtenberg 1998</a>, <a href="#ref-Jaffe_2002" role="doc-biblioref">2002</a>)</span>, along with work by Bronwyn Hall <span class="citation">(<a href="#ref-Hall_2001" role="doc-biblioref">B. Hall, Jaffe, and Trajtenberg 2001</a>; <a href="#ref-Hall_2005" role="doc-biblioref">B. H. Hall, Jaffe, and Trajtenberg 2005</a>; <a href="#ref-Hall_2012" role="doc-biblioref">B. H. Hall and Harhoff 2012</a>)</span>, Colin Webb and Helene Dernis at the OECD <span class="citation">(<a href="#ref-Webb_2005" role="doc-biblioref">Webb et al. 2005</a>)</span> and Dietmar Harhoff <span class="citation">(<a href="#ref-Harhoff_1999" role="doc-biblioref">Harhoff et al. 1999</a>; <a href="#ref-Harhoff_2003" role="doc-biblioref">Harhoff, Scherer, and Vopel 2003</a>)</span> are essential reading. For those interested in exploring the wider literature on patent citations a <a href="https://www.lens.org/lens/search/scholar/list?collectionId=203349">Lens public collection</a> is available to assist readers with getting started. The <a href="https://www.lens.org/lens/search/scholar/list?collectionId=203349">collection</a> is dynamic and will automatically update to the latest literature on patent citations.</p>
<p>In this chapter we will begin with the non-patent literature and move step by step through the issues that need to be considered when working with patent citation data. We will use examples from synthetic biology and CRISPR genome editing technology to illustrate approaches to the non-patent literature and patent citations and finish with recent research on identifying technology paths with citation data.</p>
<div id="non-patent-literature" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Non Patent Literature<a href="citations.html#non-patent-literature" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Non-patent literature mainly takes the form of scientific publications such as journals, books, or chapters but also extends to other types of materials such as manuals, news reports, drawings and websites. As we will see below, viewed from a data science perspective the non-patent literature is that it generally takes the form of messy free text that requires extensive cleaning. However, from approximately 1980 onwards NPL has become an increasingly important focus for research.</p>
<p>For those seeking to track the emergence of research on NPL citations we will simply provide some useful way points. An important starting point of research on non patent literature is work by <span class="citation">Carpenter, Cooper, and Narin (<a href="#ref-Carpenter_1980" role="doc-biblioref">1980</a>)</span> on the links between basic academic research and patent activity using US patent citation data. This was followed in the mid-1990s by more detailed studies with larger scale data by Narin and collaborators <span class="citation">(<a href="#ref-Narin_1995" role="doc-biblioref">F. Narin, Hamilton, and Olivastro 1995</a>; <a href="#ref-Narin_1997" role="doc-biblioref">Francis Narin, Hamilton, and Olivastro 1997</a>)</span>. This was accompanied by the growing proliferation of technology specific studies using citations such as work by <span class="citation">Meyer (<a href="#ref-Meyer_2000" role="doc-biblioref">2000</a>)</span> on the relationship between nanoscale technologies and the cited literature. Citation based studies have tended to be heavily focused, for reasons of accessibility, on US patent data. However, in the mid-2000s work by <span class="citation">Callaert, Looy, et al. (<a href="#ref-Callaert_2006" role="doc-biblioref">2006</a>)</span> examined 10,000 citations from the US and the European Patent Offices. This was followed by work to address the significant problems that exist with noise in NPL data using machine learning models <span class="citation">(<a href="#ref-Callaert_2011" role="doc-biblioref">Callaert, Grouwels, and Looy 2011</a>)</span>. The growing availability of NPL citation data is reflected in the growth of national and sectoral studies such as work by <span class="citation">Fukuzawa and Ida (<a href="#ref-Fukuzawa_2015" role="doc-biblioref">2015</a>)</span> exploring the links between scientific articles and patents for leading researchers in Japan. Recent work by <span class="citation">Ding et al. (<a href="#ref-Ding_2017" role="doc-biblioref">2017</a>)</span> has focused on the characteristics of scientific articles that facilitate knowledge flows between science and technology while <span class="citation">Chen (<a href="#ref-Chen_2017" role="doc-biblioref">2017</a>)</span> has explored the textual similarities between scientific articles and the contents of patent applications. Research by <span class="citation">Rizzo et al. (<a href="#ref-Rizzo_2018" role="doc-biblioref">2018</a>)</span> has focused on the closeness of publicly funded research and radical inventions in UK filings at the European Patent Office.</p>
<p>As this very brief set of way markers suggests, a significant body of work on diverse topics has emerged around the non-patent literature. While much of the original research focused on US patent data the creation of the EPO World Patent Statistical Database (PATSTAT) has made the wider non-patent literature available in a single table and served as a spur for research <span class="citation">(<a href="#ref-Webb_2005" role="doc-biblioref">Webb et al. 2005</a>; <a href="#ref-Callaert_2011" role="doc-biblioref">Callaert, Grouwels, and Looy 2011</a>; <a href="#ref-Karvonen_2013" role="doc-biblioref">Karvonen and Kässi 2013</a>)</span>. The recent PatentsView service from the USPTO now makes NPL citations available as a single table containing over 6 million raw references that can be freely downloaded. As such, data on non-patent literature citations is becoming more and more accessible for research.</p>
<p>We will consider the case of PatentsView in more detail below. However, one of the most important recent development are efforts by patent databases, led by the <a href="https://www.lens.org/">open access Lens database</a> to electronically link literature databases and patent databases together.</p>
</div>
<div id="literature-and-patent-citation-data-with-the-lens" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Literature and Patent Citation Data with the Lens<a href="citations.html#literature-and-patent-citation-data-with-the-lens" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As discussed in the chapter on the scientific literature, free electronic access to sources of the scientific literature is increasingly available through services such as Crossref, PubMed and Microsoft Academic Graph. The Lens has now developed a <code>Scholar Search</code> service that includes approximately 291.4 million scholarly works from PubMed, Crossref and Microsoft Academic. These records are then linked to citations in patent over 110 million patent documents covered by the Lens <code>Patent Search</code>. The importance of this approach is that it allows the user to navigate the non-patent literature linked to a particular record or to extract the cited literature from a record and create a collection to download and analyse. As we will also see, the recent <code>PATCITE</code> service also allows users to retrieve data on citations at scale.</p>
<p>To illustrate the possibilities opened up by combining the scientific and patent literature we will use the example of a patent search for synthetic biology and then move into exploration of the controversial subject of genome editing.</p>
<div id="retrieving-cited-literature-from-a-patent-search" class="section level4 hasAnchor" number="6.2.0.1">
<h4><span class="header-section-number">6.2.0.1</span> Retrieving Cited Literature from a Patent Search<a href="citations.html#retrieving-cited-literature-from-a-patent-search" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We will begin with a simple patent search of the titles and abstracts for terms relating to synthetic biology in the Lens.</p>
<blockquote>
<p>title: (“synthetic biology” OR (“synthetic genome” OR (“synthetic genomes” OR “synthetic genomics”))) OR abstract: (“synthetic biology” OR (“synthetic genome” OR (“synthetic genomes” OR “synthetic genomics”)))</p>
</blockquote>
<p>You can try this exact query and example live by following this <a href="https://www.lens.org/lens/scholar/search/results?patentQueryId=fv9ylnbkosQklD7F6hnaE96bpl76I3qQAJNHCqMtXc4J9Gw8nAyrsCKM9eCrn5ih&amp;preview=true">link</a>. To make the most of the Lens we recommend that you register as a user (registration is free) as this will allow you to create collections for export.</p>
<p>At the time of writing this search generated 287 patent results and 776 cited works that can be viewed in the Cited Works panel as illustrated below:</p>
<p><img src="images/citations/lenscitedworks.png" width="1607" /></p>
<p>Registered users can create a public or private collection that can be shared with others and can also download the citation data up to a maximum of 50,000 records.</p>
<p>As an alternative to creating a collection an export button is provided in the Cited Works tool bar with options to download in CSV, RIS, Bibtex and JSON formats. The CSV (comma separated values) option is particularly suited to text mining or visualisation in tables while the popular Bibtex format will be useful for creating bibliographies for researchers writing in markdown. Table <a href="citations.html#tab:importcitedworks">6.1</a> displays a small selection of the results.</p>
<table>
<caption>
<span id="tab:importcitedworks">Table 6.1: </span>Sample Literature Fields from the Lens for Synthetic Biology
</caption>
<thead>
<tr>
<th style="text-align:left;">
Title
</th>
<th style="text-align:right;">
Publication Year
</th>
<th style="text-align:left;">
DOI
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Pathway optimization and key enzyme evolution of N-acetylneuraminate biosynthesis using an in vivo aptazyme-based biosensor
</td>
<td style="text-align:right;">
2017
</td>
<td style="text-align:left;">
10.1016/j.ymben.2017.08.001
</td>
</tr>
<tr>
<td style="text-align:left;">
Transcriptome-based identification of the optimal reference CHO genes for normalisation of qPCR data
</td>
<td style="text-align:right;">
2017
</td>
<td style="text-align:left;">
10.1002/biot.201700259
</td>
</tr>
<tr>
<td style="text-align:left;">
The N-Acetylmuramic Acid 6-Phosphate Phosphatase MupP Completes the Pseudomonas Peptidoglycan Recycling Pathway Leading to Intrinsic Fosfomycin Resistance
</td>
<td style="text-align:right;">
2017
</td>
<td style="text-align:left;">
10.1128/mbio.00092-17
</td>
</tr>
<tr>
<td style="text-align:left;">
Efficient whole-cell biocatalyst for Neu5Ac production by manipulating synthetic, degradation and transmembrane pathways
</td>
<td style="text-align:right;">
2016
</td>
<td style="text-align:left;">
10.1007/s10529-016-2215-z
</td>
</tr>
<tr>
<td style="text-align:left;">
Droplet immobilization within a polymeric organogel improves lipid bilayer durability and portability
</td>
<td style="text-align:right;">
2016
</td>
<td style="text-align:left;">
10.1039/c6lc00391e
</td>
</tr>
</tbody>
</table>
<p>A wide range of other fields are also available with the downloaded data such as authors, keywords, abstracts (where available), MeSH terms (Medical Subject Headings), Chemicals, source urls and the number of patent documents that reference an article among others. This is therefore a very rich set of data for further exploration.</p>
<p>Taking our small sample data for synthetic biology we can identify the articles that are the top cited in the patent dataset as displayed in Table <a href="citations.html#tab:topranking">6.2</a>. Note that the <code>Referenced by Patent Count</code> column refers to the total known count of patent documents citing the article. This measure will therefore generally favour older and foundational literature. This is revealed by the dominance of citations to the Basic Local Alignment Search Tool (BLAST) algorithm that is very widely used in biology.</p>
<table>
<caption>
<span id="tab:topranking">Table 6.2: </span>Sample Literature Fields from the Lens for Synthetic Biology
</caption>
<thead>
<tr>
<th style="text-align:left;">
Title
</th>
<th style="text-align:right;">
Publication Year
</th>
<th style="text-align:right;">
Referenced by Patent Count
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Basic Local Alignment Search Tool
</td>
<td style="text-align:right;">
1990
</td>
<td style="text-align:right;">
8553
</td>
</tr>
<tr>
<td style="text-align:left;">
Continuous cultures of fused cells secreting antibody of predefined specificity
</td>
<td style="text-align:right;">
1975
</td>
<td style="text-align:right;">
7772
</td>
</tr>
<tr>
<td style="text-align:left;">
A general method applicable to the search for similarities in the amino acid sequence of two proteins
</td>
<td style="text-align:right;">
1970
</td>
<td style="text-align:right;">
6370
</td>
</tr>
<tr>
<td style="text-align:left;">
Gapped BLAST and PSI-BLAST: a new generation of protein database search programs
</td>
<td style="text-align:right;">
1997
</td>
<td style="text-align:right;">
5978
</td>
</tr>
<tr>
<td style="text-align:left;">
Amino acid substitution matrices from protein blocks
</td>
<td style="text-align:right;">
1992
</td>
<td style="text-align:right;">
2347
</td>
</tr>
<tr>
<td style="text-align:left;">
Melamine Deaminase and Atrazine Chlorohydrolase: 98 Percent Identical but Functionally Different
</td>
<td style="text-align:right;">
2001
</td>
<td style="text-align:right;">
1692
</td>
</tr>
<tr>
<td style="text-align:left;">
Comparison of biosequences
</td>
<td style="text-align:right;">
1981
</td>
<td style="text-align:right;">
1627
</td>
</tr>
<tr>
<td style="text-align:left;">
Rapid and efficient site-specific mutagenesis without phenotypic selection
</td>
<td style="text-align:right;">
1985
</td>
<td style="text-align:right;">
1568
</td>
</tr>
<tr>
<td style="text-align:left;">
The tac promoter: a functional hybrid derived from the trp and lac promoters
</td>
<td style="text-align:right;">
1983
</td>
<td style="text-align:right;">
1161
</td>
</tr>
<tr>
<td style="text-align:left;">
One-step inactivation of chromosomal genes in Escherichia coli K-12 using PCR products
</td>
<td style="text-align:right;">
2000
</td>
<td style="text-align:right;">
836
</td>
</tr>
</tbody>
</table>
<p>The availability of this type of data opens up a wide range of research opportunities and the maximum export of 50,000 records per query is likely to prove ample for most research purposes. These opportunities include.</p>
<ol style="list-style-type: lower-alpha">
<li>Refining search strategies by text mining the titles and abstracts and keywords of cited literature;</li>
<li>Exploring networks of patent documents citing a key piece of literature;</li>
<li>Examining available data on the funding of scientific research cited in patent documents as part of innovation studies;</li>
<li>Assessing issues around the closeness of scientific research to inventions.</li>
</ol>
<p>One illustration of patent exploration using our sample dataset might involve genome editing with CRISPR/CAS9. Patent activity in this field has attracted widespread attention following litigation between the University of California and the Harvard-MIT Broad Institute and the patent landscape linked to the CRISPR dispute is discussed in detail by <span class="citation">Egelie et al. (<a href="#ref-Egelie_2016" role="doc-biblioref">2016</a>)</span> (see also <span class="citation">Ledford (<a href="#ref-Ledford_2016" role="doc-biblioref">2016</a>)</span>, <span class="citation">Ledford (<a href="#ref-Ledford_2017" role="doc-biblioref">2017</a>)</span>, <span class="citation">Ledford (<a href="#ref-Ledford_2018" role="doc-biblioref">2018</a>)</span>). While not linked directly to patent activity, genome editing in humans using CRISPR/CAS9 has also recently attracted international media attention <span class="citation">(<a href="#ref-Cyranoski_2018" role="doc-biblioref">Cyranoski and Ledford 2018</a>)</span>.</p>
<p>Two key researchers in this field are Jennifer Doudna at Berkeley and Feng Zhang at the Harvard-MIT Broad Institute. Both appear in the NPL citations for our simple patent query for synthetic biology. By splitting the data so that each author name appears on its own row we can identify our authors of interest in the dataset as show in Table <a href="citations.html#tab:crispr">6.3</a></p>
<table>
<caption>
<span id="tab:crispr">Table 6.3: </span>CRISPR Authors Cited in Patent Literature
</caption>
<thead>
<tr>
<th style="text-align:left;">
Lens ID
</th>
<th style="text-align:left;">
Author/s
</th>
<th style="text-align:right;">
Publication Year
</th>
<th style="text-align:left;">
Title
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
<a href="https://www.lens.org/018-893-049-787-145" class="uri">https://www.lens.org/018-893-049-787-145</a>
</td>
<td style="text-align:left;">
Jennifer A. Doudna
</td>
<td style="text-align:right;">
2016
</td>
<td style="text-align:left;">
New CRISPR-Cas systems from uncultivated microbes.
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.lens.org/049-383-529-000-174" class="uri">https://www.lens.org/049-383-529-000-174</a>
</td>
<td style="text-align:left;">
Jennifer A. Doudna
</td>
<td style="text-align:right;">
2014
</td>
<td style="text-align:left;">
Enhanced homology-directed human genome engineering by controlled timing of CRISPR/Cas9 delivery.
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.lens.org/009-802-337-761-385" class="uri">https://www.lens.org/009-802-337-761-385</a>
</td>
<td style="text-align:left;">
Feng Zhang
</td>
<td style="text-align:right;">
2015
</td>
<td style="text-align:left;">
Cpf1 is a single RNA-guided endonuclease of a class 2 CRISPR-Cas system.
</td>
</tr>
<tr>
<td style="text-align:left;">
<a href="https://www.lens.org/150-951-500-543-136" class="uri">https://www.lens.org/150-951-500-543-136</a>
</td>
<td style="text-align:left;">
Feng Zhang
</td>
<td style="text-align:right;">
2013
</td>
<td style="text-align:left;">
Double Nicking by RNA-Guided CRISPR Cas9 for Enhanced Genome Editing Specificity
</td>
</tr>
</tbody>
</table>
<p>In Table <a href="citations.html#tab:crispr">6.3</a> we see four publications by the key researchers cited in the sample dataset. Clicking on one of the links provides access to details on the records and also to patent documents that cite the literature at the Lens. In Figure <a href="citations.html#fig:citedworks">6.1</a> we have used the record in the third row of Table <a href="citations.html#tab:crispr">6.3</a> above. In Figure <a href="citations.html#fig:citedworks">6.1</a> we can see the publication record and also the 170 patent documents that cite this paper.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:citedworks"></span>
<img src="images/citations/examplecited.png" alt="Patent Documents Citing A Key CRISPR Research Article" width="682" />
<p class="caption">
Figure 6.1: Patent Documents Citing A Key CRISPR Research Article
</p>
</div>
<p>This suggests that one opportunity for using the cited literature identified from a raw search is to begin to build a portfolio of documents that cite key researchers in a field such as CRISPR. In the case of the four articles identified above we simply used the hyperlink to access the data, selected Patent Citations and then View full patent data. By viewing the full patent data for each link we were able to add the citing patents to a new CRISPR collection in a few minutes (you must be logged in to create a collection). At the time of writing a collection of 304 patent documents in 256 families linked to these CRISPR articles . This collection is is publicly accessible at <a href="https://www.lens.org/lens/collection/167967">https://www.lens.org/lens/collection/167967</a>.</p>
<p>As this makes clear, linking the scientific and patent literature together makes it very easy to construct an exploratory patent portfolio in minutes. In the past this might have taken weeks or months. However, it is now also possible to work on a larger scale.</p>
</div>
</div>
<div id="retrieving-citations-at-scale-with-patcite" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Retrieving Citations at Scale with PATCITE<a href="citations.html#retrieving-citations-at-scale-with-patcite" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <a href="https://www.lens.org/lens/patcite">PATCITE tool</a> in the Lens is a recent introduction that allows a user to paste in a set of article or patent identifiers to retrieve citation data. The advantage of PATCITE is that it is possible to do this in bulk at the level of thousands of identifiers, such as the widespread Document Object Identifiers (dois) for the scientific literature. This will normally be more convenient when working with data from other datasets.</p>
<p>To briefly illustrate PATCITE we will use the four documents identified from the CRISPR publications identified above.</p>
<blockquote>
<p>“10.1038/nature21059
10.7554/elife.04766
10.1016/j.cell.2015.09.038
10.1016/j.cell.2013.08.021”</p>
</blockquote>
<p>We paste these numbers into the option to <code>Explore the cited scholarly work found in patent literature</code>. This will then produce a screen showing the four articles and a Citing Patents list. At the time of writing this list contained 304 Members. Figure <a href="citations.html#fig:patcite">6.2</a> displays the summary of results and demonstrates that it is possible to retrieve a patent portfolio based on the use of document identifiers (dois) that can then be exported (see Export results in the bottom left of Figure <a href="citations.html#fig:patcite">6.2</a>).</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:patcite"></span>
<img src="images/citations/patcite.png" alt="PATCITE Results for Four Key CRISPR Articles" width="1570" />
<p class="caption">
Figure 6.2: PATCITE Results for Four Key CRISPR Articles
</p>
</div>
<p>PATCITE also features analysis tools such as rankings of applicants citing the documents and the visualisation of networks of citations. The visualisation of networks is likely to be of particular interest as it allows for the exploration of other literature cited in a patent document. By way of illustration, Figure <a href="citations.html#fig:patcitenetwork">6.3</a> shows all patent citations linked to a literature record.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:patcitenetwork"></span>
<img src="images/citations/patcite_network.png" alt="Literature Citation Network for A CRISPR related Patent Filing" width="714" />
<p class="caption">
Figure 6.3: Literature Citation Network for A CRISPR related Patent Filing
</p>
</div>
<p>PATCITE includes options to export both the cited literature (where using patent numbers as the starting point) or citing patent documents. This is particularly useful when working at a larger scale.</p>
<p>In recent work on synthetic biology, <span class="citation">P. D. Oldham and Hall (<a href="#ref-Oldham483826" role="doc-biblioref">2018</a>)</span> mapped authors of scientific articles on synthetic biology identified in Web of Science into the global patent system by matching author names to inventor names. The literature dataset on which the research was based consisted of 4,463 publications containing 3,970 dois. These dois can be accessed <a href="https://osf.io/2xa5y/">here</a> if you would like to reproduce this test.</p>
<p>Figure <a href="citations.html#fig:patciteciting">6.4</a> reveals that PATCITE identified 893 of the 3,970 scientific article identifiers in patent documents. Figure <a href="citations.html#fig:patciteciting">6.4</a> displays the scientific records with the highest number of citing patent documents from a set of 2,349 patent families and 3,323 patent documents.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:patciteciting"></span>
<img src="images/citations/patcite_citinglit.png" alt="Patent Citations for Synthetic Biology Baseline Literature Dataset" width="1541" />
<p class="caption">
Figure 6.4: Patent Citations for Synthetic Biology Baseline Literature Dataset
</p>
</div>
<p>Each of the datasets can be downloaded in Excel format for further analysis. As this example makes clear PATCITE addresses issues of scale in exploring the relationship between the scientific literature and the patent literature. As discussed above in the case of CRISPR this opens up the possibility of creating collections of patent data based on links with the scientific literature either as a starting point for a search strategy, mapping the impacts of research, or exploring the closeness of relationships between the scientific literature and the patent literature in innovation studies.</p>
<p>In the case of new and emerging areas of science and technology, such as synthetic biology PATCITE also opens up the possibility of overcoming some of the limitations of key word based searches. In the case of synthetic biology it can be argued that it is emerging <em>within</em> the wider field of genetic engineering and biotechnology and uses much the same language. This makes it difficult to develop a keyword strategy that adequately captures the field without capturing unrelated activity. At the same time, an additional challenge with keyword strategies is that analysts through the selection of particular terms inevitably impose their own definitions on emerging fields. For example, in the case of synthetic biology should we assume that any reference to a synthetic gene, or to protein engineering or systems biology in a patent document should be treated as synthetic biology?</p>
<p>PATCITE offers the possibility of beginning these explorations directly from the scientific literature and following through into the patent literature and merits serious consideration by researchers seeking to map emerging areas of science and technology in the patent system. Specifically, literature based patent searching could provide the basis for landscape construction and also be used as part of a strategy for validating the outcomes of key word based queries.</p>
<p>One logical question for researchers seeking to match the scientific literature into patent data is the issue of data capture. That is, it is not immediately clear whether the 3,077 dois that were not identified in PATCITE were not identified because they are absent from patent data or because of limitations in capture. The answer to this question may be a combination of the two. In practice, the ability of an analyst to interrogate data capture at the database level, such as the accurate identification of dois, is likely to be limited. However, we can gain an insight into these issues using the raw citation data from the US PatentsView service.</p>
</div>
<div id="the-us-patentsview-non-patent-literature-table" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> The US PatentsView Non-Patent Literature Table<a href="citations.html#the-us-patentsview-non-patent-literature-table" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For most patent analysts the Lens literature and PATCITE service is the logical starting point for research, for the straight forward reason that it is so easy to use and can generate a targeted patent collection for exploration within a few minutes. However, this may not suit all purposes, particularly where larger scale data is required. It is also a very good idea to have an understanding of what the raw NPL looks like in understanding the strengths and limitations of different databases.</p>
<p>In the case of offline patent databases such as PATSTAT a table is available containing the non-patent literature for subscribers (also accessible through the online version of PATSTAT). However, the USPTO, through the PatentsView service makes a non-patent literature table available for download (presently as a 2.7 GB tab separated zipped file).<a href="#fn23" class="footnote-ref" id="fnref23"><sup>23</sup></a></p>
<p>Engaging with the raw non-patent literature data reveals that it is a free form text field. Table <a href="citations.html#tab:patentsview">6.4</a> shows a sample of entries from the over 6 million entries in the 2018 USPTO PatentsView non-patent literature table.</p>
<table>
<caption>
<span id="tab:patentsview">Table 6.4: </span>A sample of Non Patent Literature
</caption>
<thead>
<tr>
<th style="text-align:left;">
patent_id
</th>
<th style="text-align:left;">
text
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
9339622
</td>
<td style="text-align:left;">
English Translation of Chinese Examination Report; Application No. 2007800266164; 5 pages.
</td>
</tr>
<tr>
<td style="text-align:left;">
5013322
</td>
<td style="text-align:left;">
Surgery News-An Advertising supplement, Aug. 1, 1985, vol. 3, No. 15, Clayman Ovoid Model No. 8743 and Kratz/Johnson 7 mm Lightweight Model No. 8663, (2 pages).
</td>
</tr>
<tr>
<td style="text-align:left;">
8773357
</td>
<td style="text-align:left;">
U.S. Office Action dated Dec. 23, 2011 in U.S. Appl. No. 12/571,157.
</td>
</tr>
<tr>
<td style="text-align:left;">
7307640
</td>
<td style="text-align:left;">
Duke, “Dreamcast Technical Specs”, Sega Dreamcast Review, Sega, Feb. 1999, www.game-revolution.com.
</td>
</tr>
<tr>
<td style="text-align:left;">
8543711
</td>
<td style="text-align:left;">
Ranjan, S. and Rolia, J., Fu, H., and Knightly, E., “QoS-Driven Server Migration for Internet Data Centers,” In Proc. of IWQoS 2002, Miami, FL, May 2002.
</td>
</tr>
<tr>
<td style="text-align:left;">
5849555
</td>
<td style="text-align:left;">
" J. Hughes et al., ""How Does Pseudomonas Fluorescens, the Producing Organism of the Antibiotic Pseudomonic Acid A, Avoid Suicide?"", FEBS Letters, 122(2) pp. 322-324 (1980). "
</td>
</tr>
<tr>
<td style="text-align:left;">
8811330
</td>
<td style="text-align:left;">
Kaitz et al., “Changing the status of Subchannelization in OFDM mode,” IEEE 802.16 Broadband Wireless Access Working Group, IEEE C802.16d-03/80, IEEE, New York, New York (Nov. 13, 2003).
</td>
</tr>
<tr>
<td style="text-align:left;">
9500933
</td>
<td style="text-align:left;">
Jain et al., “Efficient Nonlinear Frequency Conversion with Maximal Atomic Coherence”, The American Physical Society, Physical Review Letters, vol. 77, No. 21, Nov. 18, 1996, pp. 4326-4329, 4 pages.
</td>
</tr>
<tr>
<td style="text-align:left;">
6653062
</td>
<td style="text-align:left;">
English Translation of Migulina.
</td>
</tr>
<tr>
<td style="text-align:left;">
8137555
</td>
<td style="text-align:left;">
Communication Relating to the Results of the Partial International Search for corresponding International Patent Application No. PCT/US2011/031412 mailed Aug. 9, 2011.
</td>
</tr>
<tr>
<td style="text-align:left;">
6855523
</td>
<td style="text-align:left;">
Biebricher, et al. (1986) Nature 321: 89-91.
</td>
</tr>
<tr>
<td style="text-align:left;">
8960456
</td>
<td style="text-align:left;">
Office Action issued Oct. 4, 2013 in U.S. Appl. No. 13/268,712 by Didehvar.
</td>
</tr>
<tr>
<td style="text-align:left;">
6576467
</td>
<td style="text-align:left;">
Feinberg et al., &amp;#8220;A Technique for Radiolabeling DNA Restruction Endonuclease Fragments to High Specific Activity,&amp;#8221;
</td>
</tr>
<tr>
<td style="text-align:left;">
8735564
</td>
<td style="text-align:left;">
Li et al; Detection of Human Papillomavirus Genotypes With Liquid Bead Microarray in Cervical Lesions of Northern Chinese Patients; Cancer Genetics and Cytogenetics, Elsevier Science Publishing, New York, NY, US; vol. 182; No. 1; Mar. 6, 2008; pp. 12-17; Abstract.
</td>
</tr>
<tr>
<td style="text-align:left;">
9440232
</td>
<td style="text-align:left;">
Fungi (Wikipedia.com accessed Jun. 3, 2013).
</td>
</tr>
<tr>
<td style="text-align:left;">
9169348
</td>
<td style="text-align:left;">
USPTO Office Action dated Sep. 9, 2008 for copending U.S. Appl. No. 11/391/571.
</td>
</tr>
<tr>
<td style="text-align:left;">
6940750
</td>
<td style="text-align:left;">
Jian-Gang Zhu et al. “Ultrahigh Density Vertical Magnetoresistive Random Access Memory (Invited),” Journal of Applied Physics, vol. 87, No. 9, May 1, 2000, pp. 6668-6673.
</td>
</tr>
<tr>
<td style="text-align:left;">
8343171
</td>
<td style="text-align:left;">
U.S. Appl. No. 60/990,062, filed Nov. 26, 2007.
</td>
</tr>
<tr>
<td style="text-align:left;">
8046478
</td>
<td style="text-align:left;">
MCL Paper Abstracts; Ahanger et al.; “A Language to Support Automatic Composition of Newscasts”; Journal of Compuoter Information Technology ; vol. 6, No. 3; 1998.
</td>
</tr>
<tr>
<td style="text-align:left;">
9763641
</td>
<td style="text-align:left;">
Ophir et al., “Elastography: Ultrasonic Estimation and Imaging of the Elastic Properties of Tissues,” Proc Instn Mech Engrs 213(Part H) (1999) 203-233.
</td>
</tr>
</tbody>
</table>
<p>As we can see from this small sample of over 6 million entries in the USPTO NPL data, the individual entry fields can reasonably be described as a messy text field. Among the issues that we encounter are partial references, spelling mistakes such as “Journal of Computer Information Technology”, abbreviations such as “Proc Instn Mech Engrs” and considerable variation in the presence of dois that will all need to be addressed to successfully extract the literature references.</p>
<p>To extract meaningful information from this table we would need to think about identifying patterns. For example, we might look for document identifiers (dois) for the scientific literature and note that most begin with <code>https:://doi.org</code>. We would then discover that the references to dois within the table are limited and might switch to using titles to cross match with other databases such as Crossref or PubMed. In short, when seeking to work with the NPL data, experimentation with regular expression based pattern matching and development of a strategy would rapidly become necessary to achieve meaningful results.</p>
<p>To illustrate this we will use the example of web addresses in this table. While our aim is not complete accuracy in the extraction of web addresses, we can illustrate the growing relevance of web sites as sources of prior art in the US patent data.</p>
<p>If we were looking for web addresses we could use an approach that detects the presence of <code>http</code> in a reference as a distinctive string. In practice this would filter this large table down to 304,590 entries containing this term. We would then do further work to identify the domains etc. using a regular expression pattern such as <code>www\\..*?\\.com</code>. We could also look at modifying the regular expression pattern to capture alternative URL endings such as <code>.org, .net</code> etc. This is certainly doable but could rapidly become quite complicated.</p>
<p>An alternative approach would be to recognise that others have worked on this kind of problem before with similar types of text data. We can therefore look at using existing solutions for this particular task rather than repeating work on writing regular expression patterns. In the case of the R programming language a solution to this problem is provided by Tyler Rinkr’s recent <code>qdapregex</code> package in R that complements his larger qualitative data analysis package <code>qdap</code>. We would install and load this package as follows.</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="citations.html#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;qdapregex&quot;</span>)</span>
<span id="cb94-2"><a href="citations.html#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qdapRegex)</span></code></pre></div>
<p><code>qdapregex</code> contains a function to extract urls from texts called <code>ex_url()</code> without needing to work on regular expressions. Here we create a new web object containing the text and extract the urls with <code>ex_url()</code> (for extract url). What we would like to do is to identify the top domain names (such as <code>google.com</code>) appearing in the references table.</p>
<p>In reality the way in which URLs are expressed in the references is quite messy and requires quite a lot of tidying up. We would probably need to do some more work to tidy up and validate the data for truly accurate results, but the code below takes us most of the way for the purposes of illustration.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="citations.html#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb95-2"><a href="citations.html#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qdapRegex)</span>
<span id="cb95-3"><a href="citations.html#cb95-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringi)</span>
<span id="cb95-4"><a href="citations.html#cb95-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-5"><a href="citations.html#cb95-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Our aim is to extract urls and then reduce to the domain</span></span>
<span id="cb95-6"><a href="citations.html#cb95-6" aria-hidden="true" tabindex="-1"></a><span class="co"># ex_url returns a list object</span></span>
<span id="cb95-7"><a href="citations.html#cb95-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-8"><a href="citations.html#cb95-8" aria-hidden="true" tabindex="-1"></a>web <span class="ot">&lt;-</span> npl<span class="sc">$</span>text <span class="sc">%&gt;%</span> </span>
<span id="cb95-9"><a href="citations.html#cb95-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ex_url</span>()</span>
<span id="cb95-10"><a href="citations.html#cb95-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-11"><a href="citations.html#cb95-11" aria-hidden="true" tabindex="-1"></a><span class="co"># process the list and return a data frame</span></span>
<span id="cb95-12"><a href="citations.html#cb95-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-13"><a href="citations.html#cb95-13" aria-hidden="true" tabindex="-1"></a>url <span class="ot">&lt;-</span> web <span class="sc">%&gt;%</span></span>
<span id="cb95-14"><a href="citations.html#cb95-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">map</span>(., <span class="st">`</span><span class="at">[[</span><span class="st">`</span>, <span class="dv">1</span>) <span class="sc">%&gt;%</span> <span class="co"># extract the first element from the list of results</span></span>
<span id="cb95-15"><a href="citations.html#cb95-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">discard</span>(., is.na) <span class="sc">%&gt;%</span> <span class="co"># drop NA for Not Available</span></span>
<span id="cb95-16"><a href="citations.html#cb95-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tibble</span>(<span class="at">url =</span> .) <span class="sc">%&gt;%</span> <span class="co"># convert to tibble</span></span>
<span id="cb95-17"><a href="citations.html#cb95-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>() <span class="sc">%&gt;%</span> <span class="co"># unnest list column</span></span>
<span id="cb95-18"><a href="citations.html#cb95-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">url =</span> <span class="fu">str_replace_all</span>(.<span class="sc">$</span>url, <span class="st">&quot;http://|https:|http|http:|:|//&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb95-19"><a href="citations.html#cb95-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">url =</span> <span class="fu">str_replace_all</span>(.<span class="sc">$</span>url, <span class="st">&quot;www.|&gt;&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb95-20"><a href="citations.html#cb95-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">domain =</span> <span class="fu">sub</span>(<span class="st">&quot;/.*&quot;</span>, <span class="st">&quot;&quot;</span>, url)) <span class="sc">%&gt;%</span> </span>
<span id="cb95-21"><a href="citations.html#cb95-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">domain =</span> <span class="fu">str_trim</span>(domain, <span class="at">side =</span> <span class="st">&quot;both&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb95-22"><a href="citations.html#cb95-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">domain =</span> stringi<span class="sc">::</span><span class="fu">stri_reverse</span>(domain)) <span class="sc">%&gt;%</span> <span class="co"># reverse string </span></span>
<span id="cb95-23"><a href="citations.html#cb95-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">domain =</span> <span class="fu">str_replace</span>(domain, <span class="st">&quot;^[.]|^,|^;&quot;</span>, <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span>  <span class="co"># remove junk</span></span>
<span id="cb95-24"><a href="citations.html#cb95-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">domain =</span> stringi<span class="sc">::</span><span class="fu">stri_reverse</span>(domain)) <span class="co"># reverse back</span></span>
<span id="cb95-25"><a href="citations.html#cb95-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-26"><a href="citations.html#cb95-26" aria-hidden="true" tabindex="-1"></a><span class="co"># count up domains and filter out blank results</span></span>
<span id="cb95-27"><a href="citations.html#cb95-27" aria-hidden="true" tabindex="-1"></a>domain <span class="ot">&lt;-</span> url <span class="sc">%&gt;%</span> </span>
<span id="cb95-28"><a href="citations.html#cb95-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">count</span>(domain, <span class="at">sort =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb95-29"><a href="citations.html#cb95-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(domain <span class="sc">!=</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p>This code parses the results down to 214,756 domains</p>
<table>
<caption>
<span id="tab:viewurllink">Table 6.5: </span>Top Domains in PatentsView Non-Patent Literature
</caption>
<thead>
<tr>
<th style="text-align:left;">
domain
</th>
<th style="text-align:right;">
n
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
web.archive.org
</td>
<td style="text-align:right;">
12973
</td>
</tr>
<tr>
<td style="text-align:left;">
en.wikipedia.org
</td>
<td style="text-align:right;">
10563
</td>
</tr>
<tr>
<td style="text-align:left;">
gsmarena.com
</td>
<td style="text-align:right;">
3962
</td>
</tr>
<tr>
<td style="text-align:left;">
ncbi.nlm.nih.gov
</td>
<td style="text-align:right;">
3277
</td>
</tr>
<tr>
<td style="text-align:left;">
ieeexplore.ieee.org
</td>
<td style="text-align:right;">
2879
</td>
</tr>
<tr>
<td style="text-align:left;">
youtube.com
</td>
<td style="text-align:right;">
2834
</td>
</tr>
<tr>
<td style="text-align:left;">
msdn.microsoft.com
</td>
<td style="text-align:right;">
1935
</td>
</tr>
<tr>
<td style="text-align:left;">
amazon.com
</td>
<td style="text-align:right;">
1819
</td>
</tr>
<tr>
<td style="text-align:left;">
microsoft.com
</td>
<td style="text-align:right;">
1562
</td>
</tr>
<tr>
<td style="text-align:left;">
citeseerx.ist.psu.edu
</td>
<td style="text-align:right;">
1506
</td>
</tr>
<tr>
<td style="text-align:left;">
w3.org
</td>
<td style="text-align:right;">
1490
</td>
</tr>
<tr>
<td style="text-align:left;">
ietf.org
</td>
<td style="text-align:right;">
1428
</td>
</tr>
<tr>
<td style="text-align:left;">
3gpp.org
</td>
<td style="text-align:right;">
1108
</td>
</tr>
<tr>
<td style="text-align:left;">
research.microsoft.com
</td>
<td style="text-align:right;">
1093
</td>
</tr>
<tr>
<td style="text-align:left;">
clinicaltrials.gov
</td>
<td style="text-align:right;">
936
</td>
</tr>
<tr>
<td style="text-align:left;">
cisco.com
</td>
<td style="text-align:right;">
922
</td>
</tr>
<tr>
<td style="text-align:left;">
google.com
</td>
<td style="text-align:right;">
920
</td>
</tr>
<tr>
<td style="text-align:left;">
tools.ietf.org
</td>
<td style="text-align:right;">
909
</td>
</tr>
<tr>
<td style="text-align:left;">
sciencedirect.com
</td>
<td style="text-align:right;">
857
</td>
</tr>
<tr>
<td style="text-align:left;">
merriam-webster.com
</td>
<td style="text-align:right;">
850
</td>
</tr>
</tbody>
</table>
<p>While this data would require further cleaning we now have a working idea of what the top web domains are across the US patent collection. In particular we can see that applicants make particular use of the Internet Archive at <a href="http://web.archive.org/">http://web.archive.org/</a> and the English language version of Wikipedia, with the third result focusing on the Global System for Mobile Communication (GSM) website GSM Arena <a href="https://www.gsmarena.com">https://www.gsmarena.com/</a>. We can also see that in some cases such as Microsoft (or Google), specific sub-domains such as the Microsoft Developers Network (MSDN) at <a href="https://msdn.microsoft.com/en-us/">https://msdn.microsoft.com/en-us/</a> are included. If we were to do further work we would want to trim these down to the respective core domain. As this suggests, the apparently simple task of extracting and ranking web domains involves more thought than might initially be suggested. However, awareness of existing tools can radically reduce the work involved.</p>
<p>In this section we have seen that access to the non-patent literature in patent databases has improved dramatically in recent years. As a result of the integration of the scientific literature and patent literature by the Lens it is now possible to enter scientific literature of interest and retrieve a patent portfolio in a matter of minutes. Similar developments are taking place among commercial providers such as the subscription based <a href="https://www.dimensions.ai/">Dimensions</a> database that applies machine learning to scientific publications, grant information, clinical trials data and patent data. It is likely that other companies will be working on similar initiatives.</p>
<p>When coupled with other developments such as non-patent literature tables in PATSTAT and access to raw non-patent literature with PatentsView it is clear that large scale analysis of the NPL literature is now possible. The ability to work with such data at scale will typically involve programmatic skills, but it is important to bear in mind that many other fields involve finding solutions to very similar problems, such as extracting URLs from texts. Investment in research on solutions to similar problems will often radically reduce the amount of work required and allow for the detailed exploration of the non-patent literature.</p>
<p>We now turn to the use of patent citations.</p>
</div>
<div id="patent-citations" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> Patent Citations<a href="citations.html#patent-citations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Patent citations are citations to other patent documents. They take the form of backward (cited) and forward (citing) citations. Backward citations, also referred to as back citations or cited patent documents, refer to earlier patent applications or grants that affect the scope of the claims of an application. Forward citations or citing documents refer to later filings of applications that are affected by the scope of the claims of the cited document.</p>
<p>Patent citations have two main sources <span class="citation">(<a href="#ref-Jaffe_2017" role="doc-biblioref">A. B. Jaffe and Rassenfosse 2017</a>; <a href="#ref-Hegde_2009" role="doc-biblioref">Hegde and Sampat 2009</a>)</span>:</p>
<ol style="list-style-type: decimal">
<li>inventors and their patent attorneys</li>
<li>patent examiners</li>
</ol>
<p>The different sources of patent citations have important implications. Specifically, the two different sources of citations may have very different motives for including a citation <span class="citation">(<a href="#ref-Webb_2005" role="doc-biblioref">Webb et al. 2005</a>)</span>. Thus, patent applicants and their attorneys will have an interest in disclosing references that have a limited impact on claims to novelty and inventive step. In the United States, and possibly other jurisdictions, applicants are expected to provide the prior art they are aware of as part of a duty of candour <span class="citation">(<a href="#ref-Webb_2005" role="doc-biblioref">Webb et al. 2005</a>; <a href="#ref-Cotropia_2013" role="doc-biblioref">Cotropia, Lemley, and Sampat 2013</a>)</span>. This may lead to practices such as seeking to draft around the prior art <span class="citation">(<a href="#ref-Cotropia_2013" role="doc-biblioref">Cotropia, Lemley, and Sampat 2013</a>)</span>. This perhaps explains why <span class="citation">Cotropia, Lemley, and Sampat (<a href="#ref-Cotropia_2013" role="doc-biblioref">2013</a>)</span> found that patent examiners typically ignore prior art provided by applicants.</p>
<p>In contrast, patent examiners can be expected to focus more closely on identify those that impact novelty and inventive step. Prior art searches by examiners are widely regarded as the highest quality of citations, because this involves a search by trained examiners for relevant prior art affecting an application. However, it is important to recognise that citation practices vary between patent offices <span class="citation">(<a href="#ref-Jaffe_2002" role="doc-biblioref">A. Jaffe and Trajtenberg 2002</a>; <a href="#ref-Webb_2005" role="doc-biblioref">Webb et al. 2005</a>)</span>. Thus, in the United States examiners are expected to list all relevant prior art while at the European Patent Office the examination guidelines stipulate that the European Search report include only the most <em>relevant</em> references <span class="citation">(<a href="#ref-Webb_2005" role="doc-biblioref">Webb et al. 2005</a>)</span>. The practical upshot of this is that citations from the USPTO will often be longer than those from the EPO. As such, it is important to be aware of the differences between patent offices in citation practices.</p>
<p>Bearing these issues in mind, <span class="citation">A. B. Jaffe and Rassenfosse (<a href="#ref-Jaffe_2017" role="doc-biblioref">2017</a>)</span> highlights that in broad terms patent citations provide insights in two main areas:</p>
<ol style="list-style-type: lower-alpha">
<li>the impact of inventions on other applicants and their economic and social value;</li>
<li>as proxies for knowledge flows and networks.</li>
</ol>
<p>We are now in a position to begin navigating patent citation networks.</p>
</div>
<div id="navigating-patent-networks" class="section level2 hasAnchor" number="6.6">
<h2><span class="header-section-number">6.6</span> Navigating Patent Networks<a href="citations.html#navigating-patent-networks" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In the discussion below we will focus on the basic issues involved in navigating patent citations using the CRISPR Cas9 genome editing cases above as our example. We will start with a fictional search of a patent database for CRISPR that identifies <a href="https://www.lens.org/lens/patent/081-370-519-314-433">EP2784162B1</a> from the Harvard-MIT Broad Institute for <code>Engineering of systems, methods and optimized guide compositions for sequence manipulation</code> by inventors Le Cong, Feng Zhang and other collaborators. We will then explore different ways of counting and navigating patent citations.</p>
<p>If you wish to explore the citations discussed below using the Lens you can access the reference document <a href="https://www.lens.org/lens/patent/081-370-519-314-433">EP2784162B1 here</a>. Note that counts of citations may vary across different databases. In the working examples below we use data from Clarivate Analytics Derwent Innovation database. Because of its greater coverage of national collections worldwide Derwent Innovation will often involve higher scores than other databases.</p>
<div id="back-citations" class="section level3 hasAnchor" number="6.6.1">
<h3><span class="header-section-number">6.6.1</span> Back citations<a href="citations.html#back-citations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Back citations are earlier patent filings (either applicants or grants) that may be listed by the applicant or an examiner during search and/or examination.</p>
<p>Table <a href="citations.html#tab:citedciting">6.6</a> shows a selection of patent documents (applications or grants) cited by EP2784162B1 and patent documents citing this document. In total for EP2784162B1 there were 40 documents in the cited table and 22 documents in the citing table.</p>
<table>
<caption>
<span id="tab:citedciting">Table 6.6: </span>EP2784162B1 Cited to Citing Relationships
</caption>
<thead>
<tr>
<th style="text-align:left;">
cited patents
</th>
<th style="text-align:left;">
publication number
</th>
<th style="text-align:left;">
citing patents
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
US4235871A
</td>
<td style="text-align:left;">
EP2784162B1
</td>
<td style="text-align:left;">
US10000772B2
</td>
</tr>
<tr>
<td style="text-align:left;">
US4774085A
</td>
<td style="text-align:left;">
EP2784162B1
</td>
<td style="text-align:left;">
US10077445B2
</td>
</tr>
<tr>
<td style="text-align:left;">
US4797368A
</td>
<td style="text-align:left;">
EP2784162B1
</td>
<td style="text-align:left;">
WO2016049163A2
</td>
</tr>
<tr>
<td style="text-align:left;">
WO2013176772A1
</td>
<td style="text-align:left;">
EP2784162B1
</td>
<td style="text-align:left;">
WO2016049251A1
</td>
</tr>
<tr>
<td style="text-align:left;">
US20130074819A1
</td>
<td style="text-align:left;">
EP2784162B1
</td>
<td style="text-align:left;">
WO2016069591A2
</td>
</tr>
<tr>
<td style="text-align:left;">
WO2014099744A1
</td>
<td style="text-align:left;">
EP2784162B1
</td>
<td style="text-align:left;">
WO2016182893A1
</td>
</tr>
</tbody>
</table>
<p>Table <a href="citations.html#tab:citedciting">6.6</a> illustrates the basic relationship between a reference document (EP2784162B1), its cited documents (e.g US4235871A) and citing documents (e.g. US10000772B2). The cited documents reference earlier patents and inform us about technologies that are informing and shaping the claims of our target document (EP2784162B1). Table <a href="citations.html#tab:citedselection">6.7</a> shows a selection of the cited documents ranked by the number of later filings that cite them (citing_count). The citing count in this case refers to the global total of citing patent documents. We can clearly see that the older documents are generally those with the highest citations. That is, while many patent documents will never attract citations, of those that do, older documents will typically attract higher citations scores than younger ones <span class="citation">(<a href="#ref-Webb_2005" role="doc-biblioref">Webb et al. 2005</a>)</span>.</p>
<table>
<caption>
<span id="tab:citedselection">Table 6.7: </span>EP2784162B1: Top Cited Patent Documents by Global Citing Count
</caption>
<thead>
<tr>
<th style="text-align:left;">
first_applicant
</th>
<th style="text-align:right;">
citing_count
</th>
<th style="text-align:right;">
cited_patent_count
</th>
<th style="text-align:left;">
publication_number
</th>
<th style="text-align:right;">
publication_year
</th>
<th style="text-align:left;">
title
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
PAPAHADJOPOULOS DEMETRIOS P
</td>
<td style="text-align:right;">
1858
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
US4235871A
</td>
<td style="text-align:right;">
1980
</td>
<td style="text-align:left;">
Method of encapsulating biologically active materials in lipid vesicles
</td>
</tr>
<tr>
<td style="text-align:left;">
Liposome Technology Inc. 
</td>
<td style="text-align:right;">
1276
</td>
<td style="text-align:right;">
4
</td>
<td style="text-align:left;">
US4837028A
</td>
<td style="text-align:right;">
1989
</td>
<td style="text-align:left;">
Liposomes with enhanced circulation time
</td>
</tr>
<tr>
<td style="text-align:left;">
Technology Unlimited Inc. 
</td>
<td style="text-align:right;">
1050
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:left;">
US4501728A
</td>
<td style="text-align:right;">
1985
</td>
<td style="text-align:left;">
Masking of liposomes from RES recognition
</td>
</tr>
<tr>
<td style="text-align:left;">
Syntex (UA.) Inc. 
</td>
<td style="text-align:right;">
960
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
US4897355A
</td>
<td style="text-align:right;">
1990
</td>
<td style="text-align:left;">
N[&amp;#969;,(&amp;#969;-1)-dialkyloxy]-and N-[&amp;#969;,(&amp;#969;-1)-dialkenyloxy]-alk-1-yl-N,N,N-tetrasubstituted ammonium lipids and uses therefor
</td>
</tr>
<tr>
<td style="text-align:left;">
Biogen Inc. 
</td>
<td style="text-align:right;">
828
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:left;">
US4873316A
</td>
<td style="text-align:right;">
1989
</td>
<td style="text-align:left;">
Isolation of exogenous recombinant proteins from the milk of transgenic mammals
</td>
</tr>
</tbody>
</table>
<p>One issue to bear in mind when working with patent citations is that there are a variety of reasons why a citation may be awarded. In some cases what may appear to be a relatively minor technical feature of a claimed invention may attract a citation from a patent document in an otherwise unrelated field. This will reflect the reality that patent claims are considered against the much wider background of the existing prior art. We will come back to this issue below. Furthermore, cited patents that have received the largest number of citations may not in fact be the closest to the new claimed invention. In this case the most important cited document is <a href="https://www.lens.org/lens/patent/076-607-252-634-172">WO2013176772A1</a> relating to RNA directed target DNA modification and modulation of transcription.<a href="#fn24" class="footnote-ref" id="fnref24"><sup>24</sup></a> Table <a href="citations.html#tab:citeddoudna">6.8</a> displays this family member filed by Jennifer Doudna and colleagues at the University of California at Berkeley with an earliest priority from the 25th of May 2012 (provisional application US201261652086P).</p>
<table>
<caption>
<span id="tab:citeddoudna">Table 6.8: </span>The Berkeley CRISPR Patent
</caption>
<thead>
<tr>
<th style="text-align:left;">
first_applicant
</th>
<th style="text-align:right;">
citing_count
</th>
<th style="text-align:right;">
cited_patent_count
</th>
<th style="text-align:left;">
publication_number
</th>
<th style="text-align:right;">
publication_year
</th>
<th style="text-align:left;">
title
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
THE REGENTS OF THE UNIVERSITY OF CALIFORNIA
</td>
<td style="text-align:right;">
307
</td>
<td style="text-align:right;">
2
</td>
<td style="text-align:left;">
WO2013176772A1
</td>
<td style="text-align:right;">
2013
</td>
<td style="text-align:left;">
METHODS AND COMPOSITIONS FOR RNA-DIRECTED TARGET DNA MODIFICATION AND FOR RNA-DIRECTED MODULATION OF TRANSCRIPTION
</td>
</tr>
</tbody>
</table>
<p>In this case we can see that a relatively young document has attracted a very significant number of citing documents relative to its age. In practice, documents that are relatively young but are attracting a significant number of citations should be an important focus of our attention. These will often either be highly original, and therefore disruptive in the technology space, and/or their claims will be broadly drawn. While older documents may attract more citations this may apply to a wider range of fields. Younger documents that are attracting citations are likely in technology terms to be closer to the reference document (EP2784162B1). This is particularly true where the cited document and the target document share the same International Patent Classification (IPC) or Cooperative Patent Classification (CPC) codes.</p>
<p>Thus, the cited document with the highest level of citations in our set <a href="https://www.lens.org/lens/patent/060-160-799-009-187">US4235871A</a> shares the C12N subclass with the reference document <a href="https://www.lens.org/lens/patent/015-360-099-978-897">EP2784162A1</a> and also shares the C12N15 group. However, it does not share a sub-group with the reference document. In contrast our Berkeley patent document <a href="https://www.lens.org/lens/patent/076-607-252-634-172">WO2013176772A1</a> shares IPC code C12N15/63 and Cooperative Patent Classification (CPC) codes C12N15/102, C12N15/113, C12N15/63, C12N15/907 and C12N9/22. The fact that the documents share a significant number of classification codes suggests technological closeness and, in this case, rivalry. The rivalrous nature of this particular case will become clearer when we consider forward citations.</p>
</div>
<div id="forward-citations" class="section level3 hasAnchor" number="6.6.2">
<h3><span class="header-section-number">6.6.2</span> Forward Citations<a href="citations.html#forward-citations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>As noted above, forward citations or citing documents are later patent filings that cite a reference document (EP2784162B1). Table <a href="citations.html#tab:citingselection">6.9</a> displays a selection of the documents citing our reference document from the Broad Institute (<a href="https://www.lens.org/lens/patent/015-360-099-978-897">EP2784162B1</a>). The first point that stands out is that this data is dominated by the Broad Institute (which is a joint Harvard and MIT entity), and MIT with the bulk of forward citations arising from other filings by the Broad Institute and MIT accounting for 17, Brigham and Womens Hospital in Boston for 2, the University of California accounting for 2 and Vertex Pharmaceuticals for 1 citation.</p>
<table>
<caption>
<span id="tab:citingselection">Table 6.9: </span>A selection of Patent Documents Citing EP2784162B1
</caption>
<thead>
<tr>
<th style="text-align:left;">
first_applicant
</th>
<th style="text-align:right;">
cited_patent_count
</th>
<th style="text-align:right;">
citing_count
</th>
<th style="text-align:left;">
publication_number
</th>
<th style="text-align:right;">
publication_year
</th>
<th style="text-align:left;">
title
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
University of California
</td>
<td style="text-align:right;">
241
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
US10000772B2
</td>
<td style="text-align:right;">
2018
</td>
<td style="text-align:left;">
Methods and compositions for RNA-directed target DNA modification and for RNA-directed modulation of transcription
</td>
</tr>
<tr>
<td style="text-align:left;">
University of California
</td>
<td style="text-align:right;">
262
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:left;">
US10077445B2
</td>
<td style="text-align:right;">
2018
</td>
<td style="text-align:left;">
Methods and compositions for RNA-directed target DNA modification and for RNA-directed modulation of transcription
</td>
</tr>
<tr>
<td style="text-align:left;">
THE BROAD INSTITUTE INC
</td>
<td style="text-align:right;">
134
</td>
<td style="text-align:right;">
8
</td>
<td style="text-align:left;">
WO2016049163A2
</td>
<td style="text-align:right;">
2016
</td>
<td style="text-align:left;">
USE AND PRODUCTION OF CHD8+/- TRANSGENIC ANIMALS WITH BEHAVIORAL PHENOTYPES CHARACTERISTIC OF AUTISM SPECTRUM DISORDER
</td>
</tr>
<tr>
<td style="text-align:left;">
THE BROAD INSTITUTE INC
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
9
</td>
<td style="text-align:left;">
WO2016049251A1
</td>
<td style="text-align:right;">
2016
</td>
<td style="text-align:left;">
DELIVERY, USE AND THERAPEUTIC APPLICATIONS OF THE CRISPR-CAS SYSTEMS AND COMPOSITIONS FOR MODELING MUTATIONS IN LEUKOCYTES
</td>
</tr>
<tr>
<td style="text-align:left;">
THE BROAD INSTITUTE INC
</td>
<td style="text-align:right;">
98
</td>
<td style="text-align:right;">
6
</td>
<td style="text-align:left;">
WO2016069591A2
</td>
<td style="text-align:right;">
2016
</td>
<td style="text-align:left;">
COMPOSITIONS, METHODS AND USE OF SYNTHETIC LETHAL SCREENING
</td>
</tr>
</tbody>
</table>
<p>An alternative way of viewing patent citations is to visualise them as a tree of relationships. Figure <a href="citations.html#fig:citingnet1">6.5</a> displays the set of citing applicants and citing documents for EP2784162B1 as a tree grouped on the first applicant.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:citingnet1"></span>
<div id="htmlwidget-6b58453a8b8024616c24" style="width:600px;height:480px;" class="collapsibleTree html-widget"></div>
<script type="application/json" data-for="htmlwidget-6b58453a8b8024616c24">{"x":{"data":{"name":".","children":[{"name":"University of California","children":[{"name":"US10000772B2"},{"name":"US10077445B2"}]},{"name":"THE BROAD INSTITUTE INC","children":[{"name":"WO2016049163A2"},{"name":"WO2016049251A1"},{"name":"WO2016069591A2"},{"name":"WO2016182893A1"},{"name":"WO2016205745A2"},{"name":"WO2017070605A1"},{"name":"WO2017074788A1"},{"name":"WO2017075451A1"},{"name":"WO2017075465A1"},{"name":"WO2017075478A2"},{"name":"WO2017219027A1"},{"name":"WO2018035250A1"},{"name":"WO2018035364A1"},{"name":"WO2018049025A2"}]},{"name":"MIT","children":[{"name":"WO2016205728A1"},{"name":"WO2017147196A1"},{"name":"WO2017161325A1"}]},{"name":"THE BRIGHAM AND WOMEN'S HOSPITAL INC","children":[{"name":"WO2017069958A2"},{"name":"WO2018067991A1"}]},{"name":"VERTEX PHARMACEUTICALS INCORPORATED","children":[{"name":"WO2018013840A1"}]}]},"options":{"hierarchy":["first_applicant","publication_number"],"input":null,"attribute":"leafCount","linkLength":220,"fontSize":10,"tooltip":false,"collapsed":false,"zoomable":false,"margin":{"top":20,"bottom":20,"left":30,"right":95},"fill":"lightsteelblue"}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 6.5: Citing Documents Grouped by Applicants
</p>
</div>
<p>In this case 9 of the 22 documents involve inventors from the Broad Institute (Feng Zhang and Le Cong) while the two citing documents from the University of California involve four of the inventors on the Berkeley document cited by the Broad Institute (<a href="https://www.lens.org/lens/patent/076-607-252-634-172">WO2013176772A1</a>).</p>
<p>There are actually two separate types of activity that emerge in the cited and citing documents for <a href="https://www.lens.org/lens/patent/015-360-099-978-897">EP2784162A1</a>.</p>
<ol style="list-style-type: lower-alpha">
<li>We observe competing filings by two groups of inventors, one from Berkeley and the other from Harvard-MIT Broad Institute. A filing from Berkeley that occurred before the Broad Institute filing is cited by the Broad Institute, but <em>later</em> filings from Berkeley cite the Broad Institute filing.</li>
<li>The Broad Institute filings are self-citing one of their own earlier filings in later filings.</li>
</ol>
<p>In the first case the citation data reveals competition in this space. The second case would however require further investigation of whether the citations are inserted by the applicants (and therefore less likely to affect the scope of the newer filings) or whether they are inserted by examiners (and may therefore impact on the scope of the claims). A review of the source of the citation in the first of the Broad Institute citing documents <a href="https://www.lens.org/181-916-033-206-149">WO2016049163A2</a> reveals that it was inserted by the applicant and has no recorded impact on the claims (typically marked by XYI). Note also that even where an earlier filing does have an impact on the claims it may be possible to identify ways to modify the claims to accommodate the prior art.</p>
<!--- follow up and insert dossier link for this record--->
<p>We now have a basic grounding in back and forward patent citations using our CRISPR case. What we have learned is that:</p>
<ol style="list-style-type: decimal">
<li>Back citations may be from a wide range of technology fields in the prior art;</li>
<li>Back citations may not have a direct impact on the scope of the later filing, particularly where they are inserted by the applicant rather than examiners;</li>
<li>Older patent documents generally attract more citations than newer ones;</li>
<li>Recent documents that are attracting a significant number of citations deserve close attention;</li>
<li>Attention is required to International Patent Classification/Cooperative Patent Classification codes to determine how close cited inventions are to each other;</li>
<li>Competition can be observed between applicants where they are citing each other;</li>
<li>The sources of citations matter. Citations by examiners have a greater impact than citations inserted by applicants.</li>
</ol>
<p>Each of these observations merits further exploration than can be provided in this chapter. The background literature discussed in this chapter explores a number of these topics. We now turn to the question of how to approach counts of patent citations and the implications of counting citations by patent families. Once again we will use the CRISPR example as a real world case.</p>
</div>
</div>
<div id="counting-citations-by-patent-families" class="section level2 hasAnchor" number="6.7">
<h2><span class="header-section-number">6.7</span> Counting Citations by Patent Families<a href="citations.html#counting-citations-by-patent-families" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As we have seen above, one method or exploring the landscape of patent citations is to focus on individual documents. However, as we will now see, conducting analysis on a per document basis may miss the majority of patent citations associated with the wider patent family and the key document or documents within a family. As such <em>counts limited to individual documents may radically underestimate the impact of a claimed invention within technology space</em>.</p>
<p>We can illustrate this for our reference document EP2784162B1 and its wider INPADOC patent family. At the time of writing this document forms part of a patent family with 277 members that has 369 cited patents and 717 citing patent documents <!--- do I mean documents here--->. Table <a href="citations.html#tab:EP2784162B1">6.10</a> displays the top ranking publications based on citing patents within this family.</p>
<table>
<caption>
<span id="tab:EP2784162B1">Table 6.10: </span>Broad Institute CRISPR Patent Family Ranked by Citing Patent Count
</caption>
<thead>
<tr>
<th style="text-align:left;">
publication_number
</th>
<th style="text-align:right;">
count_of_citing_patents
</th>
<th style="text-align:left;">
title_original
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
US8697359B1
</td>
<td style="text-align:right;">
274
</td>
<td style="text-align:left;">
CRISPR-Cas systems and methods for altering expression of gene products
</td>
</tr>
<tr>
<td style="text-align:left;">
WO2014093622A2
</td>
<td style="text-align:right;">
157
</td>
<td style="text-align:left;">
DELIVERY, ENGINEERING AND OPTIMIZATION OF SYSTEMS, METHODS AND COMPOSITIONS FOR SEQUENCE MANIPULATION AND THERAPEUTIC APPLICATIONS | DÉLIVRANCE, FABRICATION ET OPTIMISATION DE SYSTÈMES, DE PROCÉDÉS ET DE COMPOSITIONS POUR LA MANIPULATION DE SÉQUENCES ET APPLICATIONS THÉRAPEUTIQUES
</td>
</tr>
<tr>
<td style="text-align:left;">
WO2014093712A1
</td>
<td style="text-align:right;">
125
</td>
<td style="text-align:left;">
ENGINEERING OF SYSTEMS, METHODS AND OPTIMIZED GUIDE COMPOSITIONS FOR SEQUENCE MANIPULATION | FABRICATION DE SYSTÈMES, PROCÉDÉS ET COMPOSITIONS DE GUIDE OPTIMISÉES POUR LA MANIPULATION DE SÉQUENCES
</td>
</tr>
<tr>
<td style="text-align:left;">
WO2014093655A2
</td>
<td style="text-align:right;">
122
</td>
<td style="text-align:left;">
ENGINEERING AND OPTIMIZATION OF SYSTEMS, METHODS AND COMPOSITIONS FOR SEQUENCE MANIPULATION WITH FUNCTIONAL DOMAINS | FABRICATION ET OPTIMISATION DE SYSTÈMES, DE PROCÉDÉS ET DE COMPOSITIONS POUR LA MANIPULATION DE SÉQUENCE AVEC DES DOMAINES FONCTIONNELS
</td>
</tr>
<tr>
<td style="text-align:left;">
US20140179770A1
</td>
<td style="text-align:right;">
121
</td>
<td style="text-align:left;">
Delivery, Engineering and Optimization of Systems, Methods and Compositions for Sequence Manipulation and Therapeutic Applications
</td>
</tr>
</tbody>
</table>
<p>This data makes clear that our original reference document EP2784162B1, with 23 citing documents at the time of writing, was not reflecting the wider citation landscape for this patent family. Furthermore, as in the litigation surrounding this case discussed by <span class="citation">Egelie et al. (<a href="#ref-Egelie_2016" role="doc-biblioref">2016</a>)</span> and <span class="citation">Ledford (<a href="#ref-Ledford_2018" role="doc-biblioref">2018</a>)</span>, the most important individual document in this family is in fact US granted patent <a href="https://www.lens.org/lens/patent/014-343-086-528-757">US8697359B1</a>. Specifically, while, as we have seen, the Broad Institute patent family cites the Berkeley patent application, in reality the Broad Institute patent attorneys used the expedited examination procedure at the USPTO with the effect that the Broad Institute patent filing was granted <em>before</em> the Berkeley filing was examined.^[[Source: Broad Communications 2022<a href="https://www.broadinstitute.org/crispr/journalists-statement-and-background-crispr-patent-process">FOR JOURNALISTS: STATEMENTS AND BACKGROUND ON THE CRISPR PATENT PROCESS</a>, Updated February 28]</p>
<p>The important insight from a patent family based perspective is that a focus on citation counts for individual documents will often miss the wider picture. As such, it is important, wherever possible, to construct citation analysis at the patent family level. This will not only assist with identifying the most important documents in a family based on citing counts but in larger datasets will allow for the identification of the most important patent families in terms of their impact upon other actors within the technology space.</p>
<p>When working with patent citations on a family level bear in mind that the number of family members and thus of citations will depend on the family definition. Thus, simple or DOCDB families will be smaller than INPADOC families as we have seen in the earlier discussion of Patent Families. The citing landscape for those families is therefore also likely to be smaller.</p>
<p>In thinking about calculating the number of citing patent documents we might be tempted to simply sum the count of citing patents provided with the export of the data from the patent database. For this working example we would obtain an answer of 5,138 as the gross figure. However, that is not actually what we want. The figure that we want is the count of <code>distinct</code> or <code>unique</code> citing documents to arrive at the citing count for the patent family.</p>
<p>The basic process for making this calculation is to:</p>
<ol style="list-style-type: decimal">
<li>Separate out the citing patent numbers onto new rows;</li>
<li>Identify any duplicate numbers.</li>
<li>Remove or ignore the duplicates and count the distinct documents.</li>
</ol>
<p>This procedure gives the correct figure of 718 citations directly linked to this patent family as we will see in the calculation below. This calculation can be performed in a number of different ways, depending on the tools you are using, with commands such as <code>DISTINCT</code> or <code>unique()</code> and so on. The important idea is to avoid over counting the citing documents by ensuring that any duplicates of the same document are not included in the count. The code below shows a step by step approach to this type of calculation in R. There will be more efficient routes to achieving this in R and other languages. The aim here is to show the steps in a transparent way. This basically consists of separating, trimming, filtering duplicates and counting.</p>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="citations.html#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb96-2"><a href="citations.html#cb96-2" aria-hidden="true" tabindex="-1"></a>EP2784162B1 <span class="sc">%&gt;%</span> <span class="co"># table containing the citing data</span></span>
<span id="cb96-3"><a href="citations.html#cb96-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(citing_patents, <span class="at">sep =</span> <span class="st">&quot;;&quot;</span>) <span class="sc">%&gt;%</span> <span class="co"># separate citing documents onto own rows</span></span>
<span id="cb96-4"><a href="citations.html#cb96-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">citing_patents =</span> <span class="fu">str_trim</span>(citing_patents, <span class="at">side =</span> <span class="st">&quot;both&quot;</span>)) <span class="sc">%&gt;%</span> <span class="co"># trim whitespace</span></span>
<span id="cb96-5"><a href="citations.html#cb96-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">duplicated =</span> <span class="fu">duplicated</span>(citing_patents)) <span class="sc">%&gt;%</span> <span class="co"># identify duplicates</span></span>
<span id="cb96-6"><a href="citations.html#cb96-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(duplicated <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span> <span class="co"># filter to unique documents</span></span>
<span id="cb96-7"><a href="citations.html#cb96-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(citing_patents) <span class="sc">%&gt;%</span> <span class="co"># choose the column to count</span></span>
<span id="cb96-8"><a href="citations.html#cb96-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tally</span>() <span class="co"># count</span></span></code></pre></div>
<table>
<thead>
<tr>
<th style="text-align:right;">
n
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
718
</td>
</tr>
</tbody>
</table>
<p>If we were to work with a larger dataset containing multiple patent families we would therefore want to perform the above process based on each first filing (as the first family member) to arrive at the count of citing documents. Bear in mind that this choice will depend on the analytical task at hand. However, as we have seen in the case of EP2784162B1 this document is not the most important document in the family. As such, when working with patent search results it is important to move up one level to capture patent family data and then identify the most important documents in the set based on citation counts. The ability to do this will however depend in part on the options provided by the patent databases you have access to.</p>
</div>
<div id="patent-citations-by-generation" class="section level2 hasAnchor" number="6.8">
<h2><span class="header-section-number">6.8</span> Patent Citations by Generation<a href="citations.html#patent-citations-by-generation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Patent citations exist in what are often called layers or generations. Thus, a first generation forward citation is a document citing one of our reference documents such as EP2784162B1. A second generation citation is a document that cites the first generation document and so on. This is easier to appreciate where they are displayed in a tree structure. Figure <a href="citations.html#fig:citinggen">6.6</a> displays a tree map for PCT family members of <a href="https://www.lens.org/lens/patent/014-343-086-528-757">US8697359B1</a>. As the complete family set consists of 277 INPADOC family members we have restricted the set to PCT documents for ease of visibility. In the online version of this chapter the nodes can be selected and will expand or contract.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:citinggen"></span>
<div id="htmlwidget-42c5452964635205e898" style="width:800px;height:800px;" class="collapsibleTree html-widget"></div>
<script type="application/json" data-for="htmlwidget-42c5452964635205e898">{"x":{"data":{"name":".","children":[{"name":"WO2014093655A2","children":[{"name":"EP2800811B1","children":[{"name":"EP2931897B1"},{"name":"EP3138910B1"}]},{"name":"EP3138910B1","children":[{"name":"EP2928496A1"},{"name":"EP3138912A1"}]},{"name":"WO2016057961A1","children":[{"name":"EP3237611A4"},{"name":"WO2018118585A1"},{"name":"WO2018195313A1"}]}]},{"name":"WO2014093622A2","children":[{"name":"EP3009511A2","children":[{"name":"KR1897213B1"},{"name":"WO2016166340A1"},{"name":"WO2017015015A1"},{"name":"WO2017064546A1"},{"name":"WO2017188797A1"},{"name":"WO2018013720A1"},{"name":"WO2018115390A1"}]},{"name":"JP2016514479A","children":[{"name":"JP2016534125A"}]},{"name":"US9505827B2","children":[{"name":"US10064398B2"},{"name":"US10149462B2"},{"name":"US9783593B2"},{"name":"US9896516B2"},{"name":"US9924705B2"},{"name":"US9938357B2"},{"name":"US9938358B2"},{"name":"US9963716B2"}]},{"name":"WO2015040402A1","children":[{"name":"WO2015159087A1"}]},{"name":"WO2015088643A1","children":[{"name":"EP2264163B1"},{"name":"EP2787075B1"},{"name":"EP3071698A4"},{"name":"EP3381279A1"},{"name":"US9528136B2"},{"name":"US9701964B2"},{"name":"WO2016061374A1"},{"name":"WO2017079724A1"},{"name":"WO2018025206A1"},{"name":"WO2018085586A1"}]},{"name":"WO2016033246A1","children":[{"name":"WO2017155408A1"},{"name":"WO2017180711A1"},{"name":"WO2017193053A1"},{"name":"WO2018035450A1"}]},{"name":"WO2016187717A1","children":[{"name":"US10150963B2"},{"name":"US10150964B2"}]},{"name":"WO2016205711A1","children":[{"name":"WO2017152015A1"}]},{"name":"WO2017023801A1","children":[{"name":"CN107475292A"},{"name":"WO2018201144A1"}]}]},{"name":"WO2014093712A1","children":[{"name":"CN104195177A","children":[{"name":"CN104611368A"},{"name":"CN104611368B"},{"name":"CN104928321A"},{"name":"CN104928321B"}]},{"name":"EP2764103A2","children":[{"name":"US10000772B2"},{"name":"US10011849B1"},{"name":"US10017760B2"},{"name":"US10077445B2"},{"name":"US10113167B2"},{"name":"US8945839B2"},{"name":"US9822372B2"},{"name":"US9840713B2"},{"name":"US9982278B2"},{"name":"US9982279B1"},{"name":"WO2015089351A1"},{"name":"WO2015089354A1"},{"name":"WO2015089364A1"},{"name":"WO2015089419A2"},{"name":"WO2015089465A1"},{"name":"WO2015089486A2"},{"name":"WO2016049163A2"},{"name":"WO2016049251A1"},{"name":"WO2016069591A2"},{"name":"WO2016086197A1"},{"name":"WO2016094867A1"},{"name":"WO2016094872A1"},{"name":"WO2016094874A1"},{"name":"WO2016094880A1"},{"name":"WO2016100974A1"},{"name":"WO2016106236A1"},{"name":"WO2016106244A1"},{"name":"WO2016108926A1"},{"name":"WO2016120480A1"},{"name":"WO2016138488A2"},{"name":"WO2016182893A1"},{"name":"WO2016205728A1"},{"name":"WO2016205745A2"},{"name":"WO2016205764A1"},{"name":"WO2017069958A2"},{"name":"WO2017070605A1"},{"name":"WO2017074788A1"},{"name":"WO2017075294A1"},{"name":"WO2017075451A1"},{"name":"WO2017075465A1"},{"name":"WO2017075478A2"},{"name":"WO2017087708A1"},{"name":"WO2017147196A1"},{"name":"WO2017161325A1"},{"name":"WO2017219027A1"},{"name":"WO2018013840A1"},{"name":"WO2018035250A1"},{"name":"WO2018035364A1"},{"name":"WO2018049025A2"},{"name":"WO2018067991A1"},{"name":"WO2018083606A1"}]},{"name":"JP2016501036A","children":[{"name":"JP2016512048A"}]},{"name":"WO2016123230A1","children":[{"name":"CN106244591A"}]},{"name":"WO2017090761A1","children":[{"name":"WO2017183724A1"}]}]},{"name":"WO2014093595A1","children":[{"name":"WO2015089277A1","children":[{"name":"EP3105325A4"},{"name":"WO2017044419A1"},{"name":"WO2017147432A1"}]},{"name":"WO2015168800A1","children":[{"name":"CN105671083A"},{"name":"WO2018110805A1"}]},{"name":"WO2016183438A1","children":[{"name":"GB2559922A"},{"name":"WO2017070632A3"},{"name":"WO2018086623A1"}]}]},{"name":"WO2014093635A1","children":[{"name":"JP2016520318A","children":[{"name":"JP2017534294A"}]},{"name":"WO2015026885A1","children":[{"name":"DE102015006335A1"},{"name":"DE102015014252A1"},{"name":"EP3159407A1"},{"name":"GB2550745A"},{"name":"WO2016123230A1"},{"name":"WO2016164356A1"}]},{"name":"WO2015089427A1","children":[{"name":"CN105177038A"},{"name":"CN105177038B"},{"name":"EP3155116A4"},{"name":"WO2016201047A1"},{"name":"WO2017144630A1"},{"name":"WO2017205834A1"}]},{"name":"WO2015089462A1","children":[{"name":"JP06321757B2"},{"name":"JP2017079735A"},{"name":"US9982267B2"},{"name":"WO2017083852A1"},{"name":"WO2017100791A1"},{"name":"WO2017175072A1"},{"name":"WO2017190091A1"},{"name":"WO2018002783A1"},{"name":"WO2018085644A1"}]},{"name":"WO2016028682A1","children":[{"name":"GB2552861A"},{"name":"WO2017053879A1"},{"name":"WO2017189964A3"}]},{"name":"WO2016054225A1","children":[{"name":"WO2018009525A1"},{"name":"WO2018009534A1"}]},{"name":"WO2016161380A1","children":[{"name":"WO2018106727A1"}]}]},{"name":"WO2014093718A1","children":[{"name":"WO2017044776A1","children":[{"name":"WO2018195555A1"}]}]},{"name":"WO2014093694A1","children":[{"name":"US9409139B2","children":[{"name":"US10053688B2"},{"name":"US9555388B2"},{"name":"US9677067B2"},{"name":"US9833761B2"},{"name":"US9839894B2"},{"name":"US9889423B2"},{"name":"US9895673B2"},{"name":"US9981239B2"}]},{"name":"WO2016007604A1","children":[{"name":"US9403141B2"}]}]},{"name":"WO2014093661A2","children":[{"name":"EP2771468A1","children":[{"name":"EP2784162B1"},{"name":"EP2825654B1"},{"name":"EP2896697B1"},{"name":"EP2898075B1"},{"name":"EP2921557B1"},{"name":"WO2014093709A1"},{"name":"WO2014202250A1"}]},{"name":"JP2016521975A","children":[{"name":"JP2016521555A"}]}]},{"name":"WO2014093701A1","children":[{"name":"EP2982758A1","children":[{"name":"US10144933B2"},{"name":"US10149905B2"},{"name":"US9605019B2"},{"name":"US9617547B2"},{"name":"US9695211B2"},{"name":"US9885037B2"},{"name":"US9982257B2"}]},{"name":"WO2015200378A1","children":[{"name":"US9822407B2"},{"name":"US9850484B2"},{"name":"US9988674B2"},{"name":"WO2018013558A1"},{"name":"WO2018086512A1"},{"name":"WO2018129368A2"}]}]},{"name":"WO2014204727A1","children":[{"name":"WO2016196361A1","children":[{"name":"EP3307872A4"}]}]},{"name":"WO2014204724A1","children":[{"name":"US20160139124A1","children":[{"name":"US9993503B2"}]}]},{"name":"WO2014204723A1","children":[{"name":"WO2016154493A1","children":[{"name":"WO2018067886A3"},{"name":"WO2018088635A1"}]}]},{"name":"WO2014204729A1","children":[{"name":"CN104611368A","children":[{"name":"CN105861547A"}]}]},{"name":"WO2014204728A1","children":[{"name":"WO2017031232A1","children":[{"name":"US10022425B2"}]}]},{"name":"WO2014204726A1","children":[{"name":"WO2016109255A1","children":[{"name":"US9580715B2"},{"name":"US9738897B2"}]},{"name":"WO2016176191A1","children":[{"name":"WO2018154380A1"}]},{"name":"WO2016205623A1","children":[{"name":"WO2018132939A1"}]},{"name":"WO2017173054A1","children":[{"name":"WO2018213476A1"}]}]},{"name":"WO2014204727A9","children":[{"name":"CN105274095A","children":[{"name":"CN106172237A"}]}]}]},"options":{"hierarchy":["publication_number","citing_patents","citing_patents_second"],"input":null,"attribute":"leafCount","linkLength":145,"fontSize":10,"tooltip":false,"collapsed":true,"zoomable":false,"margin":{"top":20,"bottom":20,"left":30,"right":105},"fill":"lightsteelblue"}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 6.6: First and Second Generations of Citations from PCT filings for a CRISPR Family
</p>
</div>
<p>In Figure <a href="citations.html#fig:citinggen">6.6</a> the first layer consists of the PCT family members of US8697359B1. Then we see the first generation of citing patents followed by a selection of the second generation of citing patents. In both theory and practice, the citation tree could extend through multiple generations until no more generations of forward citations can be identified.</p>
<p>An alternative way of viewing a citation tree is to cluster the records on the applicant names in the different generations. To achieve accurate results you should expect to clean the applicant names before hand. Figure <a href="citations.html#fig:citapplns">6.7</a> displays the same data with applicant names.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:citapplns"></span>
<div id="htmlwidget-9cd57c727648fba316e9" style="width:800px;height:800px;" class="collapsibleTree html-widget"></div>
<script type="application/json" data-for="htmlwidget-9cd57c727648fba316e9">{"x":{"data":{"name":".","children":[{"name":"BROAD INSTITUTE INC","children":[{"name":"UNIV CALIFORNIA","children":[{"name":"BROAD INSTITUTE INC"},{"name":"SIGMA-ALDRICH CO LLC"},{"name":"TAKARA BIO USA INC"},{"name":"UNIVERSITY OF MASSACHUSETTS"},{"name":"Agenovir Corporation"}]},{"name":"SIGMA-ALDRICH CO LLC","children":[{"name":"SIGMA-ALDRICH CO LLC"}]},{"name":"EDITAS MEDICINE INC","children":[{"name":"UNIVERSITY OF MASSACHUSETTS"},{"name":"Agenovir Corporation"},{"name":"GE HEALTHCARE BIO-SCIENCES CORP"},{"name":"CARIBOU BIOSCIENCES INC"}]},{"name":"BROAD INSTITUTE INC","children":[{"name":"INSTITUTE FOR BASIC SCIENCE"},{"name":"WAGENINGEN UNIVERSITEIT"},{"name":"EMORY UNIVERSITY"},{"name":"CRISPR THERAPEUTICS AG"},{"name":"UNIV YONSEI"},{"name":"WASHINGTON UNIVERSITY"},{"name":"KEYGENE N.V."},{"name":"EDITAS MEDICINE INC"},{"name":"UNIV CALIFORNIA"},{"name":"Inscripta Inc."},{"name":"BROAD INSTITUTE INC"},{"name":"UNIV OF COLORADO"},{"name":"BRIGHAM & WOMENS HOSPITAL"},{"name":"MEIOGENIX"},{"name":"MIT"},{"name":"VERTEX PHARMACEUTICALS"},{"name":"NOVARTIS AG"},{"name":"INSTITUTE OF GENETICS AND DEVELOPMENTAL BIOLOGY CHINESE ACADEMY OF SCIENCES"},{"name":"CELLECTIS"},{"name":"UNIVERSITY OF ILLINOIS"},{"name":"OXFORD UNIVERSITY INNOVATION LIMITED"},{"name":"REPUBLIC OF KOREA NATIONAL FORENSIC SERVICE"},{"name":"FELDAN BIO INC."},{"name":"MOORE Tara"},{"name":"MASSACHUSETTS EYE AND EAR INFIRMARY"},{"name":"VINDICO NANOBIOTECHNOLOGY LLC"},{"name":"YOUHEALTH BIOTECH LIMITED"},{"name":"SIGMA-ALDRICH CO LLC"},{"name":"VOYAGER THERAPEUTICS INC"},{"name":"KOLIBREE"}]},{"name":"Kymab Limited","children":[{"name":"Kymab Limited"},{"name":"GREEN BIOLOGICS LIMITED"}]},{"name":"Regeneron Pharmaceuticals Inc","children":[{"name":"Regeneron Pharmaceuticals Inc"},{"name":"HARVARD COLLEGE"},{"name":"SNIPR TECHNOLOGIES LIMITED"},{"name":"THE JACKSON LABORATORY"},{"name":"KYOTO UNIVERSITY"},{"name":"KIEWLICH David"}]},{"name":"CARIBOU BIOSCIENCES INC","children":[{"name":"ERASMUS UNIVERSITY MEDICAL CENTER ROTTERDAM"},{"name":"EDITAS MEDICINE INC"},{"name":"WOOLF Tod M."},{"name":"EBBU LLC"}]},{"name":"EXERKINE CORPORATION","children":[{"name":"10X GENOMICS INC"}]},{"name":"UNIVERSITY OF MINNESOTA","children":[{"name":"Shandong Baifu Gene Technology Co. Ltd."},{"name":"PRECISION BIOSCIENCES INC."}]},{"name":"Nanjing University","children":[{"name":"Guangzhou Institute of Biomedicine And Health Chinese Academy of Science"},{"name":"Northwest Institute Of Plateau Biology Chinese Academy of Sciences"}]},{"name":"PIONEER HI-BRED INTERNATIONAL INC","children":[{"name":"Suzhou GenePharma Co. Ltd."},{"name":"KWS SAAT SE"},{"name":"SILENCE THERAPEUTICS"},{"name":"PIONEER HI-BRED INTERNATIONAL INC"},{"name":"STANFORD UNIVERSITY"}]},{"name":"NATIONAL UNIVERSITY CORPORATION KOBE UNIVERSITY","children":[{"name":"NATIONAL UNIVERSITY CORPORATION KOBE UNIVERSITY"}]},{"name":"UNIV LAVAL","children":[{"name":"Anhui Kedgene Biotechnology Co. Ltd."},{"name":"MGENPLUS CO. LTD."}]},{"name":"MIT","children":[{"name":"HARVARD COLLEGE"},{"name":"INSTITUTE OF GENETICS AND DEVELOPMENTAL BIOLOGY CHINESE ACADEMY OF SCIENCES"}]},{"name":"STC.UNM","children":[{"name":"UNIV JOHNS HOPKINS"}]},{"name":"UNIV TEXAS TECH","children":[{"name":"STANFORD UNIVERSITY"}]},{"name":"Twist Bioscience","children":[{"name":"Twist Bioscience"}]},{"name":"Gen9 Inc.","children":[{"name":"Twist Bioscience"}]},{"name":"Centre Hospitalier Universitaire Vaudois","children":[{"name":"SHIN NIPPON BIOMEDICAL LABORATORIES LTD."},{"name":"WAVE LIFE SCIENCES"}]},{"name":"General Hospital Corporation","children":[{"name":"General Hospital Corporation"},{"name":"LIFE TECHNOLOGIES CORPORATION"},{"name":"SHANGAHI INSTITUTES FOR BIOLOGICAL SCIENCES CHINESE ACADEMY OF SCIENCES"},{"name":"EDITAS MEDICINE INC"}]},{"name":"NORTH CAROLINA STATE UNIVERSITY","children":[{"name":"NORTH CAROLINA STATE UNIVERSITY"},{"name":"INSTITUTE OF PROCESS ENGINEERING CHINESE ACADEMY OF SCIENCES"}]},{"name":"BLOODCENTER RESEARCH FOUNDATON","children":[{"name":"Astellas Institute for Regenerative Medicine"}]},{"name":"STANFORD UNIVERSITY","children":[{"name":"NANTOMICS LLC"},{"name":"KOREA ADVANCED INSTITUTE OF SCIENCE AND TECHNOLOGY"}]},{"name":"Guangzhou Institute of Biomedicine And Health Chinese Academy of Science","children":[{"name":"HUANG Jie"}]},{"name":"MODERNATX INC.","children":[{"name":"ModernaTX Inc."}]},{"name":"UNIVERSITY OF SOUTH FLORIDA","children":[{"name":"Regeneron Pharmaceuticals Inc"}]},{"name":"UNIVERSITY OF PENNSYLVANIA","children":[{"name":"CRISPR THERAPEUTICS AG"}]},{"name":"INTELLIA THERAPEUTICS INC.","children":[{"name":"Translate Bio Inc."}]},{"name":"Qingdao Livestock Veterinarian Research Institution","children":[{"name":"Guizhou University"}]}]}]},"options":{"hierarchy":["applicant_first","citing_applicant_first","citing_applicant_second"],"input":null,"attribute":"leafCount","linkLength":145,"fontSize":10,"tooltip":false,"collapsed":false,"zoomable":false,"margin":{"top":20,"bottom":20,"left":30,"right":400},"fill":"lightsteelblue"}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 6.7: Citations from PCT filings for a CRISPR Family by Applicants
</p>
</div>
<p>The visualisation of patent citations across multiple generations draws our attention to competitive activity in the technology space for CRISPR. However, it also draws our attention to the issue of knowledge spillovers that have been a detailed focus of analysis on the scientific literature for patent citations to which we now briefly turn.</p>
</div>
<div id="citations-and-knowledge-spillovers" class="section level2 hasAnchor" number="6.9">
<h2><span class="header-section-number">6.9</span> Citations and Knowledge Spillovers<a href="citations.html#citations-and-knowledge-spillovers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Put simply, knowledge spillovers occur where the knowledge provided by an applicant informs later inventions by other unrelated applicants that may or may not be working in the same technology space.</p>
<p>One way to address knowledge spillovers is to weight the citation data to remove forward citations by the same applicant. In this case the approach taken is to identify and remove any incidence of the Broad Institute from the first generation of forward citations leaving only those affected by the reference family. Figure <a href="citations.html#fig:weighting">6.8</a> displays the impact of this approach.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:weighting"></span>
<div id="htmlwidget-775107e6328d1c886265" style="width:800px;height:800px;" class="collapsibleTree html-widget"></div>
<script type="application/json" data-for="htmlwidget-775107e6328d1c886265">{"x":{"data":{"name":".","children":[{"name":"BROAD INSTITUTE INC","children":[{"name":"UNIV CALIFORNIA","children":[{"name":"BROAD INSTITUTE INC"},{"name":"SIGMA-ALDRICH CO LLC"},{"name":"TAKARA BIO USA INC"},{"name":"UNIVERSITY OF MASSACHUSETTS"},{"name":"Agenovir Corporation"}]},{"name":"SIGMA-ALDRICH CO LLC","children":[{"name":"SIGMA-ALDRICH CO LLC"}]},{"name":"EDITAS MEDICINE INC","children":[{"name":"UNIVERSITY OF MASSACHUSETTS"},{"name":"Agenovir Corporation"},{"name":"GE HEALTHCARE BIO-SCIENCES CORP"},{"name":"CARIBOU BIOSCIENCES INC"}]},{"name":"Kymab Limited","children":[{"name":"Kymab Limited"},{"name":"GREEN BIOLOGICS LIMITED"}]},{"name":"Regeneron Pharmaceuticals Inc","children":[{"name":"Regeneron Pharmaceuticals Inc"},{"name":"HARVARD COLLEGE"},{"name":"SNIPR TECHNOLOGIES LIMITED"},{"name":"THE JACKSON LABORATORY"},{"name":"KYOTO UNIVERSITY"},{"name":"KIEWLICH David"}]},{"name":"CARIBOU BIOSCIENCES INC","children":[{"name":"ERASMUS UNIVERSITY MEDICAL CENTER ROTTERDAM"},{"name":"EDITAS MEDICINE INC"},{"name":"WOOLF Tod M."},{"name":"EBBU LLC"}]},{"name":"EXERKINE CORPORATION","children":[{"name":"10X GENOMICS INC"}]},{"name":"UNIVERSITY OF MINNESOTA","children":[{"name":"Shandong Baifu Gene Technology Co. Ltd."},{"name":"PRECISION BIOSCIENCES INC."}]},{"name":"Nanjing University","children":[{"name":"Guangzhou Institute of Biomedicine And Health Chinese Academy of Science"},{"name":"Northwest Institute Of Plateau Biology Chinese Academy of Sciences"}]},{"name":"PIONEER HI-BRED INTERNATIONAL INC","children":[{"name":"Suzhou GenePharma Co. Ltd."},{"name":"KWS SAAT SE"},{"name":"SILENCE THERAPEUTICS"},{"name":"PIONEER HI-BRED INTERNATIONAL INC"},{"name":"STANFORD UNIVERSITY"}]},{"name":"NATIONAL UNIVERSITY CORPORATION KOBE UNIVERSITY","children":[{"name":"NATIONAL UNIVERSITY CORPORATION KOBE UNIVERSITY"}]},{"name":"UNIV LAVAL","children":[{"name":"Anhui Kedgene Biotechnology Co. Ltd."},{"name":"MGENPLUS CO. LTD."}]},{"name":"STC.UNM","children":[{"name":"UNIV JOHNS HOPKINS"}]},{"name":"UNIV TEXAS TECH","children":[{"name":"STANFORD UNIVERSITY"}]},{"name":"Twist Bioscience","children":[{"name":"Twist Bioscience"}]},{"name":"Gen9 Inc.","children":[{"name":"Twist Bioscience"}]},{"name":"Centre Hospitalier Universitaire Vaudois","children":[{"name":"SHIN NIPPON BIOMEDICAL LABORATORIES LTD."},{"name":"WAVE LIFE SCIENCES"}]},{"name":"General Hospital Corporation","children":[{"name":"General Hospital Corporation"},{"name":"LIFE TECHNOLOGIES CORPORATION"},{"name":"SHANGAHI INSTITUTES FOR BIOLOGICAL SCIENCES CHINESE ACADEMY OF SCIENCES"},{"name":"EDITAS MEDICINE INC"}]},{"name":"NORTH CAROLINA STATE UNIVERSITY","children":[{"name":"NORTH CAROLINA STATE UNIVERSITY"},{"name":"INSTITUTE OF PROCESS ENGINEERING CHINESE ACADEMY OF SCIENCES"}]},{"name":"BLOODCENTER RESEARCH FOUNDATON","children":[{"name":"Astellas Institute for Regenerative Medicine"}]},{"name":"STANFORD UNIVERSITY","children":[{"name":"NANTOMICS LLC"},{"name":"KOREA ADVANCED INSTITUTE OF SCIENCE AND TECHNOLOGY"}]},{"name":"Guangzhou Institute of Biomedicine And Health Chinese Academy of Science","children":[{"name":"HUANG Jie"}]},{"name":"MODERNATX INC.","children":[{"name":"ModernaTX Inc."}]},{"name":"UNIVERSITY OF SOUTH FLORIDA","children":[{"name":"Regeneron Pharmaceuticals Inc"}]},{"name":"UNIVERSITY OF PENNSYLVANIA","children":[{"name":"CRISPR THERAPEUTICS AG"}]},{"name":"INTELLIA THERAPEUTICS INC.","children":[{"name":"Translate Bio Inc."}]},{"name":"Qingdao Livestock Veterinarian Research Institution","children":[{"name":"Guizhou University"}]}]}]},"options":{"hierarchy":["applicant_first","citing_applicant_first","citing_applicant_second"],"input":null,"attribute":"leafCount","linkLength":145,"fontSize":10,"tooltip":false,"collapsed":false,"zoomable":false,"margin":{"top":20,"bottom":20,"left":30,"right":385},"fill":"lightsteelblue"}},"evals":[],"jsHooks":[]}</script>
<p class="caption">
Figure 6.8: Detecting Knowledge Spillovers
</p>
</div>
<p>What this leaves behind therefore is other applicants. If we were to compare this family with other patent families, the applicant with the highest score in terms of knowledge spillover would <em>arguably</em> be the applicant with the patent family with the largest number of other applicants within the forward citing landscape.</p>
<!---issue that this addresses only first generation patent docs--->
<p>Note that this simple approach does not address the issue of direct competitors. As we can see in Figure <a href="citations.html#fig:weighting">6.8</a> the University of California (Berkeley) appears in the list of distinct applicants. It is arguably the case that viewed from the perspective of this applicant, they are the originators of the CRISPR technological breakthrough and therefore the source of the knowledge spillover. Depending on ones purpose, this could of course be settled through comparison of the patent families and counting the number of affected applicants. The “winner” in this case would be the applicant with the highest score.</p>
<p>An alternative, but complementary, way of thinking about how to approach knowledge spillovers would be to focus on forward citations using the International Patent Classification (IPC) or Cooperative Patent Classification (CPC). This approach would define knowledge spillovers in terms of the impacts of a claimed invention <em>outside</em> the technology space of the invention. That is, upon other areas of the patent system.</p>
<p>Figure <a href="citations.html#fig:citingipc">6.9</a> displays the IPC Subclasses for the documents in the source family (origin) and the first and second generations of citations.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:citingipc"></span>
<img src="images/citing_ipc.png" alt="IPC Subclasses by Citing Generations" width="100%" />
<p class="caption">
Figure 6.9: IPC Subclasses by Citing Generations
</p>
</div>
<p>Figure <a href="citations.html#fig:citingipc">6.9</a> reveals that the top IPC subclass is for biotechnology and genetic engineering followed by Pharmaceuticals and Medicines. As we move from the source family into the first generation of citations (x documents) and the second generation (x documents) this pattern intensifies. As we move into the first and second generation we can see that some IPC categories actually decrease, such as new breeds of animals under A01K (for animal related biotechnology) although we would probably not want to read too much into this for the recent documents.</p>
<p>As noted above, one feature of citations is that a citation may be awarded for a technical feature that is not central to the claimed invention or wider technology area. In the case of the Broad Institute CRISPR patent family the appearance of IPC subclasses for brushes and dentistry reflect this. These documents in the second generation actually focus on an invention encouraging children to brush their teeth and have no relationship with CRISPR or genetic engineering. As such, we should bear in mind that <em>apparent spillovers across generations of citations may not involve the core features of a cited invention</em>.</p>
<p>Additional factors to consider when considering citations are the sources of citations. Many analysts might privilege citations awarded by examiners with an impact on the claimed invention (marked with X or Y). This would sharpen the focus of the citing network. Other factors to consider would be that patent documents are typically awarded more than one IPC class and that IPCs therefore form clusters and networks. In addition, we have focused in this discussion on the Subclass level. While this is common, it is also somewhat crude as subclasses are commonly quite broad. Thus, a more detailed IPC based analysis would look at groups and possibly specific subgroups within biotechnology (C12N) and their clusters and networks. This issue is addressed in the chapter on classification but is not further explored here.</p>
<p>Figure <a href="citations.html#fig:citingipc">6.9</a> also usefully highlights that we can detect trends or trajectories in technology classes across the generations of citations. The proliferation of IPCs across the generations of citations also suggests that a filing, or set of filings, may be part of one or more technology paths or trajectories.</p>
<p>At a more advanced level, citation networks and classification codes can be used for the identification of technology paths or trajectories. The analysis of technology trajectories is often traced to the work of <span class="citation">Dosi (<a href="#ref-Dosi_1982" role="doc-biblioref">1982</a>)</span> following on from the work of Thomas Kuhn on the structure of scientific revolutions. Work by <span class="citation">Hummon and Dereian (<a href="#ref-Hummon_1989" role="doc-biblioref">1989</a>)</span> focused on the analysis of a set of articles and scientific events that led to the development of DNA theory. Building on work by <span class="citation">Garfield, Sher, and Torpie (<a href="#ref-Garfield_1964" role="doc-biblioref">1964</a>)</span> they used network analysis to identify the main paths between 40 events (mainly the publications of papers) that led to the identification of the structure of DNA. Whereas earlier network analysis had focused on the analysis of the nodes within the citation network <span class="citation">Hummon and Dereian (<a href="#ref-Hummon_1989" role="doc-biblioref">1989</a>)</span> focused on the <em>links</em>. Specifically, by dividing a citation network into a set of sub-graphs and calculating the strength of the links between the papers across the sub-graphs they were able to use three counts of the traversal links that revealed “define the main path through the citation network” leading to the characterization of the structure of DNA and its confirmation <span class="citation">(<a href="#ref-Hummon_1989" role="doc-biblioref">Hummon and Dereian 1989</a>)</span>. Figure <a href="citations.html#fig:mainpath">6.10</a> reproduces the results of this analysis where numbers in the network refer to specific papers and/or events in the story of DNA.</p>
<!--- need to request copyright clearance for this. Cross check on the source--->
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:mainpath"></span>
<img src="images/citations/mainpath.png" alt="The Main Path for DNA Theory" width="100%" />
<p class="caption">
Figure 6.10: The Main Path for DNA Theory
</p>
</div>
<p>The starting point for the main path revealed by these calculations was the Isolation of nucleic acid by Miescher in 1869 and ends with the first identification of codons producing amino acids and the role of RNA at number 40.</p>
<p>For our purposes, the essential issue here is that the calculation of the linkages between nodes in the network, rather than analysis of the nodes as such, can identify the key trajectory or main path in the citation network. As <span class="citation">Hummon and Dereian (<a href="#ref-Hummon_1989" role="doc-biblioref">1989</a>)</span> points out this is similar to the calculation of the centrality of nodes in a network (which is based on calculating the shortest distance that needs to be traversed between nodes in a network).</p>
<p>Main path analysis has more recently been applied to patent citations. Here we will focus on the work of Christopher Magee at MIT and his collaborators whose recent work provides a good overview of the topic. In a review of existing work on main path analysis in areas such as Fuel Cells, Local Area Networks and the miniaturization of semi conductors, among others, Magee and collaborators observe that existing approaches run the risk of dropping important patents that contribute to the emergence of a technology field [ref]. This issue is also reflected in the problem that main path analysis typically identifies <em>one</em> main path when in practice there may be a range of paths representing contributing domains along with sub-domains within with a main path that lead to transformations (discontinuities).</p>
<p>The method demonstrated by Magee and collaborators consists of the following basic steps</p>
<ol style="list-style-type: lower-alpha">
<li><p>collecting a set of patents for the technology domain. This step involves retrieving patents specific to a recognizable body of knowledge using key words, applicants or inventors. A classification overlap method (using the acronym COM) is then used to identify highly relevant documents. Essentially this step consists of identifying documents that share overlapping classifications between the now defunct United States Patent Classification (UPC) and the International Patent Classification or, in recent work, the Cooperative Patent Classification (CPC) <span class="citation">(<a href="#ref-Park_2017" role="doc-biblioref">Park and Magee 2017</a>; <a href="#ref-Magee_2018" role="doc-biblioref">Magee et al. 2018</a>; <a href="#ref-Benson_2012" role="doc-biblioref">Benson and Magee 2012</a>, <a href="#ref-Benson_2014" role="doc-biblioref">2014</a>)</span>.</p></li>
<li><p>generating the knowledge network by retrieving the back and forward citations from the initial reference set identified above. <!---When viewed in terms of Figure \@ref(tab:citingipc), here we would be examining those documents in the back cited and citing network that share the same classification (albeit as a cruder description of the Magee et. al. methods [@Benson_2012; @Benson_2014])---></p></li>
<li><p>Measuring knowledge persistence. Magee et al argue that this is the key step in overcoming the limitations of other approaches to main path analysis using patent citation data <span class="citation">(<a href="#ref-Park_2017" role="doc-biblioref">Park and Magee 2017</a>)</span>. Highly persistent patents (that they call HPPs) are citations of patents that persist across multiple generations (layers) of citations in the backward and forward network. By searching both backwards and forwards they also argue that the problem of missing other paths can be overcome. They explain the concept of knowledge persistence as follows:</p></li>
</ol>
<blockquote>
<p>” The main concept of knowledge persistence is that a new invention is created by the recombination of existing pieces of knowledge and so, similar to Mendelian genetic inheritance, a proportion of knowledge in a patent is incorporated in its descendant patents. Therefore, in the patent system, cited and citing patents can be interpreted as ancestors and descendants from the genetic inheritance perspective.” <span class="citation">(<a href="#ref-Park_2017" role="doc-biblioref">Park and Magee 2017</a>)</span>:5</p>
</blockquote>
<ol start="4" style="list-style-type: lower-alpha">
<li>Tracing main paths from high persistence patents. This actually consists of the calculation of the high persistence patents at the level of layers (generations) by retrieving the forward and backward sets of each patent and across the network (global). <!--- greater clarity needed here on local and global---></li>
</ol>
<p>An example of the type of analysis that results from these steps is reproduced in Figure <a href="citations.html#fig:solar">6.11</a> for the case of Solar Photovoltaic Systems <span class="citation">(<a href="#ref-Park_2017" role="doc-biblioref">Park and Magee 2017</a>)</span>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:solar"></span>
<img src="images/citations/journal.pone.0170895.g006.PNG" alt="Main Patent Path Analysis Results for Solar Photovoltaic Systems" width="1386" />
<p class="caption">
Figure 6.11: Main Patent Path Analysis Results for Solar Photovoltaic Systems
</p>
</div>
<p>When we consider the raw network of citations in the top left of Figure <a href="citations.html#fig:solar">6.11</a> and the two main paths revealed in the network, it becomes clear that one of the purposes of main path analysis is to reduce the complexity of citation networks by extracting the main path(s) that involve high persistence patents in the specific technology domain.</p>
<p>In recent work this type of analysis has also been applied to trace the history of the emergence of CRISPR as a breakthrough technology from underlying genome engineering technology <span class="citation">(<a href="#ref-Magee_2018" role="doc-biblioref">Magee et al. 2018</a>)</span>. Main path analysis led them to identify three main paths,</p>
<ol style="list-style-type: lower-alpha">
<li>for cloning and restriction endonucleases,</li>
<li>for core genome editing and,</li>
<li>for endonucleases and related enzymes.</li>
</ol>
<p>Figure <a href="citations.html#fig:coreediting">6.12</a> shows the main paths in the genome engineering data leading to genome editing.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:coreediting"></span>
<img src="images/citations/journalpone0198541g006.PNG" alt="Core Genome Editing Path" width="2200" />
<p class="caption">
Figure 6.12: Core Genome Editing Path
</p>
</div>
<p>In Figure <a href="citations.html#fig:coreediting">6.12</a> the term RE refers to Restriction Endonuclease. What this figure tells us, is that the analysis reveals that there are six clusters of activity that make up the CRISPR main path. The colours represent the technology cluster and the numbers represent the actual patent numbers. The key features of this path are that they represent advances in synthetic restriction endonucleases using zinc finger nucleases (ZFN) and transcriptor activator like effector nucleases (TALENs). These patents in relation to genome editing in the field of genome engineering are then followed, and contribute to the rise of CRISPR. <span class="citation">Magee et al. (<a href="#ref-Magee_2018" role="doc-biblioref">2018</a>)</span> use a separate set of CRISPR roots (based on CPC classifiers) to explore the roots of the CRISPR patents and their overlap with the earlier genome engineering patent activity presented above.</p>
<p>Main path analysis in the case of patent citations has emerged as an area of research for technology trajectories over the last decade or so. As the work above reveals the method continues to be refined to more accurately capture paths that contribute to an emerging breakthrough and the technological sub-clusters within an emerging technology area. The particular strength of this approach is that it reduces the complexity of citation networks and makes it easier to identify the most important paths and clusters within the network. However, a possible weakness of this method is the dependence of the classification overlap method which depends on the US classification (discontinued in 2015) and the IPC/CPC. It remains to be seen whether the use of classification codes would be as robust using purely IPC or CPC codes as a basis for selection.</p>
<p>Nevertheless, in drawing attention to main path analysis our purpose is to highlight that citation analysis combined with classification and citation metrics is an important field of research that increasingly promises to make navigating citation networks significantly easier. Here it is important to recall that the task of the patent analyst is ultimately to recognise complexity but also reduce that complexity to accurate information that can be communicated to the relevant audience. Main path analysis could potentially become an important feature of the analytical tool kit by identifying the main clusters and way markers influencing the trajectory of a technology area. For this reason it deserves closer critical attention.</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Benson_2012" class="csl-entry">
Benson, Christopher L., and Christopher L. Magee. 2012. <span>“A Hybrid Keyword and Patent Class Methodology for Selecting Relevant Sets of Patents for a Technological Field.”</span> <em>Scientometrics</em> 96 (1): 69–82. <a href="https://doi.org/10.1007/s11192-012-0930-3">https://doi.org/10.1007/s11192-012-0930-3</a>.
</div>
<div id="ref-Benson_2014" class="csl-entry">
———. 2014. <span>“Technology Structural Implications from the Extension of a Patent Search Method.”</span> <em>Scientometrics</em> 102 (3): 1965–85. <a href="https://doi.org/10.1007/s11192-014-1493-2">https://doi.org/10.1007/s11192-014-1493-2</a>.
</div>
<div id="ref-Callaert_2011" class="csl-entry">
Callaert, Julie, Joris Grouwels, and Bart Van Looy. 2011. <span>“Delineating the Scientific Footprint in Technology: Identifying Scientific Publications Within Non-Patent References.”</span> <em>Scientometrics</em> 91 (2): 383–98. <a href="https://doi.org/10.1007/s11192-011-0573-9">https://doi.org/10.1007/s11192-011-0573-9</a>.
</div>
<div id="ref-Callaert_2006" class="csl-entry">
Callaert, Julie, Bart Van Looy, Arnold Verbeek, Koenraad Debackere, and Bart Thijs. 2006. <span>“Traces of Prior Art: An Analysis of Non-Patent References Found in Patent Documents.”</span> <em>Scientometrics</em> 69 (1): 3–20. <a href="https://doi.org/10.1007/s11192-006-0135-8">https://doi.org/10.1007/s11192-006-0135-8</a>.
</div>
<div id="ref-Carpenter_1980" class="csl-entry">
Carpenter, Mark P., Martin Cooper, and Francis Narin. 1980. <span>“Linkage Between Basic Research Literature and Patents.”</span> <em>Research Management</em> 23 (2): 30–35. <a href="https://doi.org/10.1080/00345334.1980.11756595">https://doi.org/10.1080/00345334.1980.11756595</a>.
</div>
<div id="ref-Chen_2017" class="csl-entry">
Chen, Lixin. 2017. <span>“Do Patent Citations Indicate Knowledge Linkage? The Evidence from Text Similarities Between Patents and Their Citations.”</span> <em>Journal of Informetrics</em> 11 (1): 63–79. <a href="https://doi.org/10.1016/j.joi.2016.04.018">https://doi.org/10.1016/j.joi.2016.04.018</a>.
</div>
<div id="ref-Cotropia_2013" class="csl-entry">
Cotropia, Christopher A., Mark A. Lemley, and Bhaven Sampat. 2013. <span>“Do Applicant Patent Citations Matter?”</span> <em>Research Policy</em> 42 (4): 844–54. <a href="https://doi.org/10.1016/j.respol.2013.01.003">https://doi.org/10.1016/j.respol.2013.01.003</a>.
</div>
<div id="ref-Cyranoski_2018" class="csl-entry">
Cyranoski, David, and Heidi Ledford. 2018. <span>“Genome-Edited Baby Claim Provokes International Outcry.”</span> <em>Nature</em> 563 (7733): 607–8. <a href="https://doi.org/10.1038/d41586-018-07545-0">https://doi.org/10.1038/d41586-018-07545-0</a>.
</div>
<div id="ref-Ding_2017" class="csl-entry">
Ding, Cherng G., Wen-Chi Hung, Meng-Che Lee, and Hung-Jui Wang. 2017. <span>“Exploring Paper Characteristics That Facilitate the Knowledge Flow from Science to Technology.”</span> <em>Journal of Informetrics</em> 11 (1): 244–56. <a href="https://doi.org/10.1016/j.joi.2016.12.004">https://doi.org/10.1016/j.joi.2016.12.004</a>.
</div>
<div id="ref-Dosi_1982" class="csl-entry">
Dosi, Giovanni. 1982. <span>“Technological Paradigms and Technological Trajectories.”</span> <em>Research Policy</em> 11 (3): 147–62. <a href="https://doi.org/10.1016/0048-7333(82)90016-6">https://doi.org/10.1016/0048-7333(82)90016-6</a>.
</div>
<div id="ref-Egelie_2016" class="csl-entry">
Egelie, Knut J, Gregory D Graff, Sabina P Strand, and Berit Johansen. 2016. <span>“The Emerging Patent Landscape of <span>CRISPR</span><span></span>cas Gene Editing Technology.”</span> <em>Nature Biotechnology</em> 34 (10): 1025–31. <a href="https://doi.org/10.1038/nbt.3692">https://doi.org/10.1038/nbt.3692</a>.
</div>
<div id="ref-Fukuzawa_2015" class="csl-entry">
Fukuzawa, Naomi, and Takanori Ida. 2015. <span>“Science Linkages Between Scientific Articles and Patents for Leading Scientists in the Life and Medical Sciences Field: The Case of Japan.”</span> <em>Scientometrics</em> 106 (2): 629–44. <a href="https://doi.org/10.1007/s11192-015-1795-z">https://doi.org/10.1007/s11192-015-1795-z</a>.
</div>
<div id="ref-Garfield_1964" class="csl-entry">
Garfield, Eugene, Irving H. Sher, and Richard J. Torpie. 1964. <span>“The Use of Citation Data in Writing the History of Science.”</span> Defense Technical Information Center. <a href="https://doi.org/10.21236/ad0466578">https://doi.org/10.21236/ad0466578</a>.
</div>
<div id="ref-Hall_2012" class="csl-entry">
Hall, Bronwyn H., and Dietmar Harhoff. 2012. <span>“Recent Research on the Economics of Patents.”</span> <em>Annual Review of Economics</em> 4 (1): 541–65. <a href="https://doi.org/10.1146/annurev-economics-080511-111008">https://doi.org/10.1146/annurev-economics-080511-111008</a>.
</div>
<div id="ref-Hall_2005" class="csl-entry">
Hall, Bronwyn H., Adam Jaffe, and Manuel Trajtenberg. 2005. <span>“Market Value and Patent Citations.”</span> <em>The RAND Journal of Economics</em> 36 (1): 16–38. <a href="http://www.jstor.org/stable/1593752">http://www.jstor.org/stable/1593752</a>.
</div>
<div id="ref-Hall_2001" class="csl-entry">
Hall, Bronwyn, Adam Jaffe, and Manuel Trajtenberg. 2001. <span>“The <span>NBER</span> Patent Citation Data File: Lessons, Insights and Methodological Tools.”</span> National Bureau of Economic Research. <a href="https://doi.org/10.3386/w8498">https://doi.org/10.3386/w8498</a>.
</div>
<div id="ref-Harhoff_1999" class="csl-entry">
Harhoff, Dietmar, Francis Narin, F. M. Scherer, and Katrin Vopel. 1999. <span>“Citation Frequency and the Value of Patented Inventions.”</span> <em>The Review of Economics and Statistics</em> 81 (3): 511–15. <a href="http://www.jstor.org/stable/2646773">http://www.jstor.org/stable/2646773</a>.
</div>
<div id="ref-Harhoff_2003" class="csl-entry">
Harhoff, Dietmar, Frederic M Scherer, and Katrin Vopel. 2003. <span>“Citations, Family Size, Opposition and the Value of Patent Rights.”</span> <em>Research Policy</em> 32 (8): 1343–63. <a href="https://doi.org/10.1016/s0048-7333(02)00124-5">https://doi.org/10.1016/s0048-7333(02)00124-5</a>.
</div>
<div id="ref-Hegde_2009" class="csl-entry">
Hegde, Deepak, and Bhaven Sampat. 2009. <span>“Examiner Citations, Applicant Citations, and the Private Value of Patents.”</span> <em>Economics Letters</em> 105 (3): 287–89. <a href="https://doi.org/10.1016/j.econlet.2009.08.019">https://doi.org/10.1016/j.econlet.2009.08.019</a>.
</div>
<div id="ref-Hummon_1989" class="csl-entry">
Hummon, Norman P., and Patrick Dereian. 1989. <span>“Connectivity in a Citation Network: The Development of DNA Theory.”</span> <em>Social Networks</em> 11 (1): 39–63. <a href="https://doi.org/10.1016/0378-8733(89)90017-8">https://doi.org/10.1016/0378-8733(89)90017-8</a>.
</div>
<div id="ref-Jaffe_2017" class="csl-entry">
Jaffe, Adam B., and Gaétan de Rassenfosse. 2017. <span>“Patent Citation Data in Social Science Research: Overview and Best Practices.”</span> <em>Journal of the Association for Information Science and Technology</em> 68 (6): 1360–74. <a href="https://doi.org/10.1002/asi.23731">https://doi.org/10.1002/asi.23731</a>.
</div>
<div id="ref-Jaffe_1998" class="csl-entry">
Jaffe, Adam, and Manuel Trajtenberg. 1998. <span>“International Knowledge Flows: Evidence from Patent Citations.”</span> National Bureau of Economic Research. <a href="https://doi.org/10.3386/w6507">https://doi.org/10.3386/w6507</a>.
</div>
<div id="ref-Jaffe_2002" class="csl-entry">
———. 2002. <em>Patents, Citations, and Innovations: A Window on the Knowledge Economy</em>. MIT Press.
</div>
<div id="ref-Karvonen_2013" class="csl-entry">
Karvonen, Matti, and Tuomo Kässi. 2013. <span>“Patent Citations as a Tool for Analysing the Early Stages of Convergence.”</span> <em>Technological Forecasting and Social Change</em> 80 (6): 1094–1107. <a href="https://doi.org/10.1016/j.techfore.2012.05.006">https://doi.org/10.1016/j.techfore.2012.05.006</a>.
</div>
<div id="ref-Ledford_2016" class="csl-entry">
Ledford, Heidi. 2016. <span>“Bitter Fight over <span>CRISPR</span> Patent Heats Up.”</span> <em>Nature</em> 529 (7586): 265–65. <a href="https://doi.org/10.1038/nature.2015.17961">https://doi.org/10.1038/nature.2015.17961</a>.
</div>
<div id="ref-Ledford_2017" class="csl-entry">
———. 2017. <span>“Broad Institute Wins Bitter Battle over <span>CRISPR</span> Patents.”</span> <em>Nature</em> 542 (7642): 401–1. <a href="https://doi.org/10.1038/nature.2017.21502">https://doi.org/10.1038/nature.2017.21502</a>.
</div>
<div id="ref-Ledford_2018" class="csl-entry">
———. 2018. <span>“Pivotal <span>CRISPR</span> Patent Battle Won by Broad Institute.”</span> <em>Nature</em>, September. <a href="https://doi.org/10.1038/d41586-018-06656-y">https://doi.org/10.1038/d41586-018-06656-y</a>.
</div>
<div id="ref-Magee_2018" class="csl-entry">
Magee, Christopher L., Patrick W. Kleyn, Brendan M. Monks, Ulrich Betz, and Subarna Basnet. 2018. <span>“Pre-Existing Technological Core and Roots for the <span>CRISPR</span> Breakthrough.”</span> Edited by Shuang-yong Xu. <em><span>PLOS</span> <span>ONE</span></em> 13 (9): e0198541. <a href="https://doi.org/10.1371/journal.pone.0198541">https://doi.org/10.1371/journal.pone.0198541</a>.
</div>
<div id="ref-Meyer_2000" class="csl-entry">
Meyer, Martin. 2000. <span>“Does Science Push Technology? Patents Citing Scientific Literature.”</span> <em>Research Policy</em> 29 (3): 409–34. <a href="https://doi.org/10.1016/s0048-7333(99)00040-2">https://doi.org/10.1016/s0048-7333(99)00040-2</a>.
</div>
<div id="ref-Narin_1995" class="csl-entry">
Narin, F., K. S. Hamilton, and D. Olivastro. 1995. <span>“Linkage Between Agency-Supported Research and Patented Industrial Technology.”</span> <em>Research Evaluation</em> 5 (3): 183–87. <a href="https://doi.org/10.1093/rev/5.3.183">https://doi.org/10.1093/rev/5.3.183</a>.
</div>
<div id="ref-Narin_1997" class="csl-entry">
Narin, Francis, Kimberly S. Hamilton, and Dominic Olivastro. 1997. <span>“The Increasing Linkage Between u.s. Technology and Public Science.”</span> <em>Research Policy</em> 26 (3): 317–30. <a href="https://doi.org/10.1016/s0048-7333(97)00013-9">https://doi.org/10.1016/s0048-7333(97)00013-9</a>.
</div>
<div id="ref-Oldham483826" class="csl-entry">
Oldham, Paul D, and Stephen Hall. 2018. <span>“Synthetic Biology: Mapping the Patent Landscape.”</span> <em>bioRxiv</em>. <a href="https://doi.org/10.1101/483826">https://doi.org/10.1101/483826</a>.
</div>
<div id="ref-Park_2017" class="csl-entry">
Park, Hyunseok, and Christopher L. Magee. 2017. <span>“Tracing Technological Development Trajectories: A Genetic Knowledge Persistence-Based Main Path Approach.”</span> Edited by Zhong-Ke Gao. <em><span>PLOS</span> <span>ONE</span></em> 12 (1): e0170895. <a href="https://doi.org/10.1371/journal.pone.0170895">https://doi.org/10.1371/journal.pone.0170895</a>.
</div>
<div id="ref-Rizzo_2018" class="csl-entry">
Rizzo, Ugo, Nicolò Barbieri, Laura Ramaciotti, and Demian Iannantuono. 2018. <span>“The Division of Labour Between Academia and Industry for the Generation of Radical Inventions.”</span> <em>The Journal of Technology Transfer</em>, August. <a href="https://doi.org/10.1007/s10961-018-9688-y">https://doi.org/10.1007/s10961-018-9688-y</a>.
</div>
<div id="ref-Scotchmer_1991" class="csl-entry">
Scotchmer, Suzanne. 1991. <span>“Standing on the Shoulders of Giants: Cumulative Research and the Patent Law.”</span> <em>Journal of Economic Perspectives</em> 5 (1): 29–41. <a href="https://doi.org/10.1257/jep.5.1.29">https://doi.org/10.1257/jep.5.1.29</a>.
</div>
<div id="ref-Webb_2005" class="csl-entry">
Webb, Colin, Hélène Dernis, Dietmar Harhoff, and Karin Hoisl. 2005. <span>“Analysing European and International Patent Citations: A Set of EPO Patent Database Building Blocks.”</span> OECD Science, Technology and Industry Working Papers 2005/9. OECD Publishing. <a href="https://EconPapers.repec.org/RePEc:oec:stiaaa:2005/9-en">https://EconPapers.repec.org/RePEc:oec:stiaaa:2005/9-en</a>.
</div>
</div>
<div class="footnotes">
<hr />
<ol start="23">
<li id="fn23"><p>To download the zipped file: [<a href="http://s3.amazonaws.com/data-patentsview-org/20180528/download/otherreference.tsv.zip" class="uri">http://s3.amazonaws.com/data-patentsview-org/20180528/download/otherreference.tsv.zip</a>][<a href="http://s3.amazonaws.com/data-patentsview-org/20180528/download/otherreference.tsv.zip" class="uri">http://s3.amazonaws.com/data-patentsview-org/20180528/download/otherreference.tsv.zip</a>]<a href="citations.html#fnref23" class="footnote-back">↩︎</a></p></li>
<li id="fn24"><p><span class="citation">Egelie et al. (<a href="#ref-Egelie_2016" role="doc-biblioref">2016</a>)</span> refer to this document by its family member US20140068797.<a href="citations.html#fnref24" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="classification.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="textmining.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["handbook.pdf", "handbook.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"show_codefolding_buttons": true
});
});
</script>

</body>

</html>
